// carbon-components-react/es/components/ContentSwitcher/next/ContentSwitcher-story.js
/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

/* eslint-disable storybook/story-exports */
import React from 'react';
import { ContentSwitcher, ContentTabs, ContentTab, ContentPanels, ContentPanel } from './';
export default {
  title: 'Experimental/unstable_ContentSwitcher',
  component: ContentSwitcher,
  subcomponents: {
    ContentTabs: ContentTabs,
    ContentTab: ContentTab,
    ContentPanels: ContentPanels,
    ContentPanel: ContentPanel
  },
  includeStories: []
};
export var Default = function Default() {
  return /*#__PURE__*/React.createElement(ContentSwitcher, null, /*#__PURE__*/React.createElement(ContentTabs, {
    activation: "manual",
    "aria-label": "Example content switcher"
  }, /*#__PURE__*/React.createElement(ContentTab, null, "Tab 1"), /*#__PURE__*/React.createElement(ContentTab, null, "Tab 2"), /*#__PURE__*/React.createElement(ContentTab, null, "Tab 3")), /*#__PURE__*/React.createElement(ContentPanels, null, /*#__PURE__*/React.createElement(ContentPanel, null, "Panel 1"), /*#__PURE__*/React.createElement(ContentPanel, null, "Panel 2"), /*#__PURE__*/React.createElement(ContentPanel, null, "Panel 3")));
};
