// carbon-components-react/es/components/DatePicker/next/DatePicker.stories.js
import _extends from "@babel/runtime/helpers/extends";

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from 'react';
import { action } from '@storybook/addon-actions';
import DatePicker from './DatePicker';
import DatePickerSkeleton from '../DatePicker.Skeleton';
import DatePickerInput from '../../DatePickerInput';
import { Layer } from '../../Layer';
var props = {
  datePicker: function datePicker() {
    return {
      dateFormat: 'm/d/Y',
      id: 'date-picker',
      light: false,
      datePickerType: 'single',
      minDate: '11/15/2020',
      maxDate: '11/01/2040',
      disable: ['07/15/2021'],
      onClose: action('onClose'),
      onOpen: action('onOpen')
    };
  },
  datePickerInput: function datePickerInput() {
    return {
      id: 'date-picker-input-id',
      className: 'some-class',
      size: undefined,
      labelText: 'Date Picker label',
      pattern: 'd{1,2}/d{4}',
      placeholder: 'mm/dd/yyyy',
      disabled: false,
      invalid: false,
      invalidText: 'A valid date is required',
      warn: false,
      warnText: 'Selected dates may cause conflicts',
      iconDescription: 'Icon description'
    };
  }
};
export default {
  title: 'Components/DatePicker',
  component: DatePicker,
  argTypes: {
    size: {
      options: ['sm', 'md', 'lg'],
      control: {
        type: 'select'
      },
      table: {
        category: 'DatePickerInput'
      }
    }
  },
  args: {
    size: 'md'
  },
  subcomponents: {
    DatePickerInput: DatePickerInput,
    DatePickerSkeleton: DatePickerSkeleton
  }
};
export var Simple = function Simple(args) {
  return /*#__PURE__*/React.createElement(DatePicker, _extends({
    datePickerType: "simple"
  }, args), /*#__PURE__*/React.createElement(DatePickerInput, _extends({
    placeholder: "mm/dd/yyyy",
    labelText: "Date Picker label",
    id: "date-picker-simple"
  }, args)));
};
export var SingleWithCalendar = function SingleWithCalendar(args) {
  return /*#__PURE__*/React.createElement(DatePicker, _extends({
    datePickerType: "single"
  }, args), /*#__PURE__*/React.createElement(DatePickerInput, _extends({
    placeholder: "mm/dd/yyyy",
    labelText: "Date Picker label",
    id: "date-picker-single"
  }, args)));
};
export var RangeWithCalendar = function RangeWithCalendar(args) {
  return /*#__PURE__*/React.createElement(DatePicker, _extends({
    datePickerType: "range"
  }, args), /*#__PURE__*/React.createElement(DatePickerInput, _extends({
    id: "date-picker-input-id-start",
    placeholder: "mm/dd/yyyy",
    labelText: "Start date"
  }, args)), /*#__PURE__*/React.createElement(DatePickerInput, _extends({
    id: "date-picker-input-id-finish",
    placeholder: "mm/dd/yyyy",
    labelText: "End date"
  }, args)));
};
export var SimpleWithLayer = function SimpleWithLayer(args) {
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(DatePicker, _extends({
    datePickerType: "simple"
  }, args), /*#__PURE__*/React.createElement(DatePickerInput, _extends({
    placeholder: "mm/dd/yyyy",
    labelText: "Date Picker label",
    id: "date-picker-simple"
  }, args))), /*#__PURE__*/React.createElement(Layer, args, /*#__PURE__*/React.createElement(DatePicker, _extends({
    datePickerType: "simple"
  }, args), /*#__PURE__*/React.createElement(DatePickerInput, _extends({
    placeholder: "mm/dd/yyyy",
    labelText: "Date Picker label",
    id: "date-picker-simple"
  }, args))), /*#__PURE__*/React.createElement(Layer, args, /*#__PURE__*/React.createElement(DatePicker, _extends({
    datePickerType: "simple"
  }, args), /*#__PURE__*/React.createElement(DatePickerInput, _extends({
    placeholder: "mm/dd/yyyy",
    labelText: "Date Picker label",
    id: "date-picker-simple"
  }, args))))));
};
export var SingleWithCalendarWithLayer = function SingleWithCalendarWithLayer(args) {
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(DatePicker, _extends({
    datePickerType: "single"
  }, args), /*#__PURE__*/React.createElement(DatePickerInput, _extends({
    placeholder: "mm/dd/yyyy",
    labelText: "Date Picker label",
    id: "date-picker-single"
  }, args))), /*#__PURE__*/React.createElement(Layer, args, /*#__PURE__*/React.createElement(DatePicker, _extends({
    datePickerType: "single"
  }, args), /*#__PURE__*/React.createElement(DatePickerInput, _extends({
    placeholder: "mm/dd/yyyy",
    labelText: "Date Picker label",
    id: "date-picker-single"
  }, args))), /*#__PURE__*/React.createElement(Layer, args, /*#__PURE__*/React.createElement(DatePicker, _extends({
    datePickerType: "single"
  }, args), /*#__PURE__*/React.createElement(DatePickerInput, _extends({
    placeholder: "mm/dd/yyyy",
    labelText: "Date Picker label",
    id: "date-picker-single"
  }, args))))));
};
export var RangeWithCalendarWithLayer = function RangeWithCalendarWithLayer(args) {
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(DatePicker, _extends({
    datePickerType: "range"
  }, args), /*#__PURE__*/React.createElement(DatePickerInput, _extends({
    id: "date-picker-input-id-start",
    placeholder: "mm/dd/yyyy",
    labelText: "Start date"
  }, args)), /*#__PURE__*/React.createElement(DatePickerInput, _extends({
    id: "date-picker-input-id-finish",
    placeholder: "mm/dd/yyyy",
    labelText: "End date"
  }, args))), /*#__PURE__*/React.createElement(Layer, args, /*#__PURE__*/React.createElement(DatePicker, _extends({
    datePickerType: "range"
  }, args), /*#__PURE__*/React.createElement(DatePickerInput, _extends({
    id: "date-picker-input-id-start",
    placeholder: "mm/dd/yyyy",
    labelText: "Start date"
  }, args)), /*#__PURE__*/React.createElement(DatePickerInput, _extends({
    id: "date-picker-input-id-finish",
    placeholder: "mm/dd/yyyy",
    labelText: "End date"
  }, args))), /*#__PURE__*/React.createElement(Layer, args, /*#__PURE__*/React.createElement(DatePicker, _extends({
    datePickerType: "range"
  }, args), /*#__PURE__*/React.createElement(DatePickerInput, _extends({
    id: "date-picker-input-id-start",
    placeholder: "mm/dd/yyyy",
    labelText: "Start date"
  }, args)), /*#__PURE__*/React.createElement(DatePickerInput, _extends({
    id: "date-picker-input-id-finish",
    placeholder: "mm/dd/yyyy",
    labelText: "End date"
  }, args))))));
};
/* eslint-disable react/prop-types */

export var DatePickerPlayground = function DatePickerPlayground() {
  return /*#__PURE__*/React.createElement(DatePicker, props.datePicker(), /*#__PURE__*/React.createElement(DatePickerInput, props.datePickerInput()), props.datePicker().datePickerType === 'range' && /*#__PURE__*/React.createElement(DatePickerInput, _extends({}, props.datePickerInput(), {
    id: "date-picker-input-2"
  })));
};
/* eslint-enable react/prop-types */

export var Skeleton = function Skeleton() {
  return /*#__PURE__*/React.createElement(DatePickerSkeleton, {
    range: true
  });
};
