// carbon-components-react/es/components/ProgressIndicator/ProgressIndicator-story.js
/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from 'react';
import { withKnobs, number, boolean, text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import { ProgressIndicator, ProgressStep } from '../ProgressIndicator';
import ProgressIndicatorSkeleton from '../ProgressIndicator/ProgressIndicator.Skeleton';
import Tooltip from '../Tooltip';
import { settings } from 'carbon-components';
import mdx from './ProgressIndicator.mdx';
var prefix = settings.prefix;
export default {
  title: 'Components/ProgressIndicator',
  component: ProgressIndicator,
  decorators: [withKnobs],
  parameters: {
    docs: {
      page: mdx
    }
  },
  subcomponents: {
    ProgressStep: ProgressStep
  }
};
export var Default = function Default() {
  return /*#__PURE__*/React.createElement(ProgressIndicator, {
    vertical: boolean('Vertical (vertical)', false),
    currentIndex: number('Current progress (currentIndex)', 1),
    spaceEqually: boolean('Space Equally (spaceEqually)', false)
  }, /*#__PURE__*/React.createElement(ProgressStep, {
    label: text('Label (label)', 'First step'),
    description: "Step 1: Getting started with Carbon Design System",
    secondaryLabel: "Optional label"
  }), /*#__PURE__*/React.createElement(ProgressStep, {
    label: "Second step with tooltip",
    description: "Step 2: Getting started with Carbon Design System",
    renderLabel: function renderLabel() {
      return /*#__PURE__*/React.createElement(Tooltip, {
        direction: "bottom",
        showIcon: false,
        triggerClassName: "".concat(prefix, "--progress-label"),
        triggerText: 'Second step with tooltip',
        tooltipId: "tooltipId-0"
      }, /*#__PURE__*/React.createElement("p", null, "Overflow tooltip content."));
    }
  }), /*#__PURE__*/React.createElement(ProgressStep, {
    label: "Third step with tooltip",
    description: "Step 3: Getting started with Carbon Design System",
    renderLabel: function renderLabel() {
      return /*#__PURE__*/React.createElement(Tooltip, {
        direction: "bottom",
        showIcon: false,
        triggerClassName: "".concat(prefix, "--progress-label"),
        triggerText: 'Third step with tooltip',
        tooltipId: "tooltipId-1"
      }, /*#__PURE__*/React.createElement("p", null, "Lorem ipsum dolor, sit amet consectetur adipisicing elit. Animi consequuntur hic ratione aliquid cupiditate, nesciunt saepe iste blanditiis cumque maxime tenetur veniam est illo deserunt sint quae pariatur. Laboriosam, consequatur."));
    }
  }), /*#__PURE__*/React.createElement(ProgressStep, {
    label: "Fourth step",
    description: "Step 4: Getting started with Carbon Design System",
    invalid: true,
    secondaryLabel: "Example invalid step"
  }), /*#__PURE__*/React.createElement(ProgressStep, {
    label: "Fifth step",
    description: "Step 5: Getting started with Carbon Design System",
    disabled: true
  }));
};
export var Interactive = function Interactive() {
  return /*#__PURE__*/React.createElement(ProgressIndicator, {
    currentIndex: number('Current progress (currentIndex)', 1),
    onChange: action('onChange')
  }, /*#__PURE__*/React.createElement(ProgressStep, {
    label: "Click me",
    description: "Step 1: Register a onChange event"
  }), /*#__PURE__*/React.createElement(ProgressStep, {
    label: "Really long label",
    description: "The progress indicator will listen for clicks on the steps"
  }), /*#__PURE__*/React.createElement(ProgressStep, {
    label: "Tooltip and really long label",
    description: "The progress indicator will listen for clicks on the steps",
    renderLabel: function renderLabel() {
      return /*#__PURE__*/React.createElement(Tooltip, {
        direction: "bottom",
        showIcon: false,
        triggerClassName: "".concat(prefix, "--progress-label"),
        triggerText: "Tooltip and really long label",
        tooltipId: "tooltipId-1"
      }, /*#__PURE__*/React.createElement("p", null, "Lorem ipsum dolor, sit amet consectetur adipisicing elit. Animi consequuntur hic ratione aliquid cupiditate, nesciunt saepe iste blanditiis cumque maxime tenetur veniam est illo deserunt sint quae pariatur. Laboriosam, consequatur."));
    }
  }));
};
Interactive.storyName = 'interactive';
export var Skeleton = function Skeleton() {
  return /*#__PURE__*/React.createElement(ProgressIndicatorSkeleton, null);
};
Skeleton.storyName = 'skeleton';
