// carbon-components-react/lib/components/ToggleSmall/ToggleSmall-story.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _addonActions = require("@storybook/addon-actions");

var _addonKnobs = require("@storybook/addon-knobs");

var _ToggleSmall = _interopRequireDefault(require("../ToggleSmall"));

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
var toggleProps = function toggleProps() {
  return {
    labelText: (0, _addonKnobs.text)('Label toggle input control (labelText)', 'Toggle element label'),
    className: 'some-class',
    labelA: (0, _addonKnobs.text)('Label for untoggled state (labelA)', ''),
    labelB: (0, _addonKnobs.text)('Label for toggled state (labelB)', ''),
    disabled: (0, _addonKnobs.boolean)('Disabled (disabled)', false),
    onChange: (0, _addonActions.action)('onChange'),
    onToggle: (0, _addonActions.action)('onToggle')
  };
};

var _default = {
  title: 'Deprecated/ToggleSmall',
  component: _ToggleSmall.default,
  decorators: [_addonKnobs.withKnobs]
};
exports.default = _default;

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("h4", null, "This component has been deprecated, please use the `size` prop provided by Toggle instead"), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement(_ToggleSmall.default, (0, _extends2.default)({
    defaultToggled: true
  }, toggleProps(), {
    className: "some-class",
    id: "toggle-1"
  })));
};

exports.Default = Default;
Default.storyName = 'toggled';
Default.parameters = {
  info: {
    text: "\n        Toggles are controls that are used to quickly switch between two possible states. The example below shows\n        an uncontrolled Toggle component. To use the Toggle component as a controlled component, set the toggled property.\n        Setting the toggled property will allow you to change the value dynamically, whereas setting the defaultToggled\n        prop will only set the value initially. This example has defaultToggled set to true. Small toggles may be used\n        when there is not enough space for a regular sized toggle. This issue is most commonly found in tables.\n      "
  }
};
