// carbon-components-react/lib/components/Toolbar/Toolbar-story.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _addonActions = require("@storybook/addon-actions");

var _iconsReact = require("@carbon/icons-react");

var _Toolbar = _interopRequireWildcard(require("../Toolbar"));

var _OverflowMenu = _interopRequireDefault(require("../OverflowMenu"));

var _OverflowMenuItem = _interopRequireDefault(require("../OverflowMenuItem"));

var _Checkbox = _interopRequireDefault(require("../Checkbox"));

var _RadioButton = _interopRequireDefault(require("../RadioButton"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
var toolbarProps = {
  className: 'some-class'
};
var inputProps = {
  className: 'some-class',
  onChange: (0, _addonActions.action)('onChange')
};
var _default = {
  title: 'Deprecated/Toolbar',
  component: _Toolbar.default,
  subcomponents: {
    ToolbarItem: _Toolbar.ToolbarItem,
    ToolbarTitle: _Toolbar.ToolbarTitle,
    ToolbarOption: _Toolbar.ToolbarOption,
    ToolbarDivider: _Toolbar.ToolbarDivider
  }
};
exports.default = _default;

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_Toolbar.default, (0, _extends2.default)({}, toolbarProps, {
    className: "some-class"
  }), /*#__PURE__*/_react.default.createElement(_Toolbar.ToolbarItem, {
    type: "search",
    placeHolderText: "Search"
  }), /*#__PURE__*/_react.default.createElement(_Toolbar.ToolbarItem, null, /*#__PURE__*/_react.default.createElement(_OverflowMenu.default, {
    renderIcon: _iconsReact.Filter16
  }, /*#__PURE__*/_react.default.createElement(_Toolbar.ToolbarTitle, {
    title: "FILTER BY"
  }), /*#__PURE__*/_react.default.createElement(_Toolbar.ToolbarOption, null, /*#__PURE__*/_react.default.createElement(_Checkbox.default, (0, _extends2.default)({}, inputProps, {
    id: "opt-1",
    labelText: "Filter option 1"
  }))), /*#__PURE__*/_react.default.createElement(_Toolbar.ToolbarOption, null, /*#__PURE__*/_react.default.createElement(_Checkbox.default, (0, _extends2.default)({}, inputProps, {
    id: "opt-2",
    labelText: "Filter option 2"
  }))), /*#__PURE__*/_react.default.createElement(_Toolbar.ToolbarOption, null, /*#__PURE__*/_react.default.createElement(_Checkbox.default, (0, _extends2.default)({}, inputProps, {
    id: "opt-3",
    labelText: "Filter option 3"
  }))))), /*#__PURE__*/_react.default.createElement(_Toolbar.ToolbarItem, null, /*#__PURE__*/_react.default.createElement(_OverflowMenu.default, null, /*#__PURE__*/_react.default.createElement(_OverflowMenuItem.default, {
    itemText: "Refresh table"
  }), /*#__PURE__*/_react.default.createElement(_Toolbar.ToolbarDivider, null), /*#__PURE__*/_react.default.createElement(_Toolbar.ToolbarTitle, {
    title: "ROW HEIGHT"
  }), /*#__PURE__*/_react.default.createElement(_Toolbar.ToolbarOption, null, /*#__PURE__*/_react.default.createElement(_RadioButton.default, (0, _extends2.default)({}, inputProps, {
    value: "short",
    id: "radio-1",
    name: "toolbar-radio",
    labelText: "Short"
  }))), /*#__PURE__*/_react.default.createElement(_Toolbar.ToolbarOption, null, /*#__PURE__*/_react.default.createElement(_RadioButton.default, (0, _extends2.default)({}, inputProps, {
    value: "tall",
    id: "radio-2",
    name: "toolbar-radio",
    labelText: "Tall"
  }))))));
};

exports.Default = Default;
Default.parameters = {
  info: {
    text: "\n      Toolbar stuff\n    "
  }
};
