// carbon-components-react/lib/components/OverflowMenu/next/OverflowMenu.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.RenderCustomIcon = exports.Default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _OverflowMenu = _interopRequireDefault(require("../OverflowMenu"));

var _OverflowMenuItem = _interopRequireDefault(require("../../OverflowMenuItem"));

var _OverflowMenu2 = _interopRequireDefault(require("../OverflowMenu.mdx"));

var _iconsReact = require("@carbon/icons-react");

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
var _default = {
  title: 'Components/OverflowMenu',
  component: _OverflowMenu.default,
  argTypes: {
    size: {
      options: ['sm', 'md', 'lg'],
      control: {
        type: 'select'
      }
    }
  },
  args: {
    size: 'md'
  },
  subcomponents: {
    OverflowMenuItem: _OverflowMenuItem.default
  },
  parameters: {
    docs: {
      page: _OverflowMenu2.default
    }
  }
};
exports.default = _default;

var Default = function Default(args) {
  return /*#__PURE__*/_react.default.createElement(_OverflowMenu.default, args, /*#__PURE__*/_react.default.createElement(_OverflowMenuItem.default, {
    itemText: "Stop app"
  }), /*#__PURE__*/_react.default.createElement(_OverflowMenuItem.default, {
    itemText: "Restart app"
  }), /*#__PURE__*/_react.default.createElement(_OverflowMenuItem.default, {
    itemText: "Rename app"
  }), /*#__PURE__*/_react.default.createElement(_OverflowMenuItem.default, {
    itemText: "Edit routes and access",
    requireTitle: true
  }), /*#__PURE__*/_react.default.createElement(_OverflowMenuItem.default, {
    hasDivider: true,
    isDelete: true,
    itemText: "Delete app"
  }));
};

exports.Default = Default;
Default.story = {
  name: 'OverflowMenu'
};

var RenderCustomIcon = function RenderCustomIcon(args) {
  return /*#__PURE__*/_react.default.createElement(_OverflowMenu.default, (0, _extends2.default)({
    renderIcon: _iconsReact.Filter16
  }, args), /*#__PURE__*/_react.default.createElement(_OverflowMenuItem.default, {
    itemText: "Filter A"
  }), /*#__PURE__*/_react.default.createElement(_OverflowMenuItem.default, {
    itemText: "Filter B"
  }));
};

exports.RenderCustomIcon = RenderCustomIcon;
