// carbon-components-react/lib/components/Button/next/Button.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports._Default = exports.Tertiary = exports.Skeleton = exports.SetOfButtons = exports.Secondary = exports.IconButton = exports.Ghost = exports.Danger = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _addonActions = require("@storybook/addon-actions");

var _iconsReact = require("@carbon/icons-react");

var _ = _interopRequireWildcard(require("../"));

var _ButtonSet = _interopRequireDefault(require("../../ButtonSet"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
var _default = {
  title: 'Components/Button',
  component: _.default,
  subcomponents: {
    ButtonSet: _ButtonSet.default,
    ButtonSkeleton: _.ButtonSkeleton
  },
  argTypes: {
    kind: {
      options: ['primary', 'secondary', 'tertiary', 'ghost', 'danger', 'danger--tertiary', 'danger--ghost'],
      control: {
        type: 'select'
      }
    },
    size: {
      options: ['sm', 'md', 'lg', 'xl', '2xl'],
      control: {
        type: 'select'
      }
    },
    children: {
      control: false
    },
    renderIcon: {
      control: false
    },
    as: {
      control: false
    },
    small: {
      table: {
        disable: true
      }
    }
  }
};
exports.default = _default;

var _Default = function _Default(args) {
  return /*#__PURE__*/_react.default.createElement(_.default, args, "Button");
};

exports._Default = _Default;

var Secondary = function Secondary(args) {
  return /*#__PURE__*/_react.default.createElement(_.default, (0, _extends2.default)({
    kind: "secondary"
  }, args), "Button");
};

exports.Secondary = Secondary;

var Tertiary = function Tertiary(args) {
  return /*#__PURE__*/_react.default.createElement(_.default, (0, _extends2.default)({
    kind: "tertiary"
  }, args), "Button");
};

exports.Tertiary = Tertiary;

var Danger = function Danger(args) {
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_.default, (0, _extends2.default)({
    kind: "danger"
  }, args), "Button"), "\xA0", /*#__PURE__*/_react.default.createElement(_.default, (0, _extends2.default)({
    kind: "danger--tertiary"
  }, args), "Tertiary Danger Button"), "\xA0", /*#__PURE__*/_react.default.createElement(_.default, (0, _extends2.default)({
    kind: "danger--ghost"
  }, args), "Ghost Danger Button"));
};

exports.Danger = Danger;

var Ghost = function Ghost(args) {
  return /*#__PURE__*/_react.default.createElement(_.default, (0, _extends2.default)({
    kind: "ghost"
  }, args), "Button");
};

exports.Ghost = Ghost;

var IconButton = function IconButton(args) {
  return /*#__PURE__*/_react.default.createElement(_.default, (0, _extends2.default)({
    renderIcon: _iconsReact.Add16,
    iconDescription: "Icon Description",
    hasIconOnly: true,
    onClick: (0, _addonActions.action)('onClick')
  }, args));
};

exports.IconButton = IconButton;

var SetOfButtons = function SetOfButtons(args) {
  return /*#__PURE__*/_react.default.createElement(_ButtonSet.default, null, /*#__PURE__*/_react.default.createElement(_.default, (0, _extends2.default)({
    kind: "secondary"
  }, args), "Secondary button"), /*#__PURE__*/_react.default.createElement(_.default, (0, _extends2.default)({
    kind: "primary"
  }, args), "Primary button"));
};

exports.SetOfButtons = SetOfButtons;

var Skeleton = function Skeleton() {
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_.ButtonSkeleton, null), "\xA0", /*#__PURE__*/_react.default.createElement(_.ButtonSkeleton, {
    size: "sm"
  }));
};

exports.Skeleton = Skeleton;
