// carbon-components-react/lib/components/ComposedModal/next/ComposedModal.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithStateManager = exports.Playground = exports.PassiveModal = exports.Default = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _reactDom = _interopRequireDefault(require("react-dom"));

var _addonActions = require("@storybook/addon-actions");

var _addonKnobs = require("@storybook/addon-knobs");

var _ComposedModal = _interopRequireWildcard(require("./ComposedModal"));

var _ModalHeader = require("./ModalHeader");

var _ModalFooter = require("./ModalFooter");

var _Select = _interopRequireDefault(require("../../Select"));

var _SelectItem = _interopRequireDefault(require("../../SelectItem"));

var _TextInput = _interopRequireDefault(require("../../TextInput"));

var _Button = _interopRequireDefault(require("../../Button"));

var _ComposedModal2 = _interopRequireDefault(require("../ComposedModal.mdx"));

var _excluded = ["size", "numberOfButtons"];

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

var sizes = {
  'Extra small (xs)': 'xs',
  'Small (sm)': 'sm',
  'Medium (md)': 'md',
  'Large (lg)': 'lg'
};
var buttons = {
  'None (0)': '0',
  'One (1)': '1',
  'Two (2)': '2',
  'Three (3)': '3'
};
var props = {
  composedModal: function composedModal() {
    return {
      numberOfButtons: (0, _addonKnobs.optionsKnob)('Number of Buttons', buttons, '2', {
        display: 'inline-radio'
      }),
      open: (0, _addonKnobs.boolean)('Open (open in <ComposedModal>)', true),
      onKeyDown: (0, _addonActions.action)('onKeyDown'),
      selectorPrimaryFocus: (0, _addonKnobs.text)('Primary focus element selector (selectorPrimaryFocus)', '[data-modal-primary-focus]'),
      size: (0, _addonKnobs.select)('Size (size)', sizes, 'md'),
      preventCloseOnClickOutside: (0, _addonKnobs.boolean)('Prevent closing on click outside of modal (preventCloseOnClickOutside)', true)
    };
  },
  modalHeader: function modalHeader() {
    var _ref = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
        titleOnly = _ref.titleOnly;

    return {
      label: (0, _addonKnobs.text)('Optional Label (label in <ModalHeader>)', 'Label'),
      title: (0, _addonKnobs.text)('Optional title (title in <ModalHeader>)', titleOnly ? "\n      Passive modal title as the message. Should be direct and 3 lines or less.\n    ".trim() : 'Modal heading'),
      iconDescription: (0, _addonKnobs.text)('Close icon description (iconDescription in <ModalHeader>)', 'Close'),
      buttonOnClick: (0, _addonActions.action)('buttonOnClick')
    };
  },
  modalBody: function modalBody() {
    return {
      hasScrollingContent: (0, _addonKnobs.boolean)('Modal contains scrollable content (hasScrollingContent)', false),
      'aria-label': (0, _addonKnobs.text)('ARIA label for content', 'Example modal content')
    };
  },
  modalFooter: function modalFooter(numberOfButtons) {
    var secondaryButtons = function secondaryButtons() {
      switch (numberOfButtons) {
        case '2':
          return {
            secondaryButtonText: (0, _addonKnobs.text)('Secondary button text (secondaryButtonText in <ModalFooter>)', 'Secondary button')
          };

        case '3':
          return {
            secondaryButtons: (0, _addonKnobs.object)('Secondary button config array (secondaryButtons)', [{
              buttonText: 'Keep both',
              onClick: (0, _addonActions.action)('onClick')
            }, {
              buttonText: 'Rename',
              onClick: (0, _addonActions.action)('onClick')
            }])
          };

        default:
          return null;
      }
    };

    return _objectSpread(_objectSpread({
      danger: (0, _addonKnobs.boolean)('Primary button danger (danger)', false),
      primaryButtonText: (0, _addonKnobs.text)('Primary button text (primaryButtonText in <ModalFooter>)', 'Primary button'),
      primaryButtonDisabled: (0, _addonKnobs.boolean)('Primary button disabled (primaryButtonDisabled in <ModalFooter>)', false)
    }, secondaryButtons(numberOfButtons)), {}, {
      onRequestClose: function onRequestClose() {
        return (0, _addonActions.action)('onRequestClose')();
      },
      onRequestSubmit: function onRequestSubmit() {
        return (0, _addonActions.action)('onRequestSubmit')();
      }
    });
  }
};

var scrollingContent = /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean id accumsan augue. Phasellus consequat augue vitae tellus tincidunt posuere. Curabitur justo urna, consectetur vel elit iaculis, ultrices condimentum risus. Nulla facilisi. Etiam venenatis molestie tellus. Quisque consectetur non risus eu rutrum.", ' '), /*#__PURE__*/_react.default.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean id accumsan augue. Phasellus consequat augue vitae tellus tincidunt posuere. Curabitur justo urna, consectetur vel elit iaculis, ultrices condimentum risus. Nulla facilisi. Etiam venenatis molestie tellus. Quisque consectetur non risus eu rutrum.", ' '), /*#__PURE__*/_react.default.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean id accumsan augue. Phasellus consequat augue vitae tellus tincidunt posuere. Curabitur justo urna, consectetur vel elit iaculis, ultrices condimentum risus. Nulla facilisi. Etiam venenatis molestie tellus. Quisque consectetur non risus eu rutrum.", ' '), /*#__PURE__*/_react.default.createElement("h3", null, "Lorem ipsum"), /*#__PURE__*/_react.default.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean id accumsan augue. Phasellus consequat augue vitae tellus tincidunt posuere. Curabitur justo urna, consectetur vel elit iaculis, ultrices condimentum risus. Nulla facilisi. Etiam venenatis molestie tellus. Quisque consectetur non risus eu rutrum.", ' '), /*#__PURE__*/_react.default.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean id accumsan augue. Phasellus consequat augue vitae tellus tincidunt posuere. Curabitur justo urna, consectetur vel elit iaculis, ultrices condimentum risus. Nulla facilisi. Etiam venenatis molestie tellus. Quisque consectetur non risus eu rutrum.", ' '), /*#__PURE__*/_react.default.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean id accumsan augue. Phasellus consequat augue vitae tellus tincidunt posuere. Curabitur justo urna, consectetur vel elit iaculis, ultrices condimentum risus. Nulla facilisi. Etiam venenatis molestie tellus. Quisque consectetur non risus eu rutrum.", ' '), /*#__PURE__*/_react.default.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean id accumsan augue. Phasellus consequat augue vitae tellus tincidunt posuere. Curabitur justo urna, consectetur vel elit iaculis, ultrices condimentum risus. Nulla facilisi. Etiam venenatis molestie tellus. Quisque consectetur non risus eu rutrum.", ' '));

var _default = {
  title: 'Components/ComposedModal',
  component: _ComposedModal.default,
  subcomponents: {
    ModalHeader: _ModalHeader.ModalHeader,
    ModalBody: _ComposedModal.ModalBody,
    ModalFooter: _ModalFooter.ModalFooter
  },
  decorators: [_addonKnobs.withKnobs],
  parameters: {
    docs: {
      page: _ComposedModal2.default
    }
  }
};
exports.default = _default;

var Playground = function Playground() {
  var _props$composedModal = props.composedModal(),
      size = _props$composedModal.size,
      numberOfButtons = _props$composedModal.numberOfButtons,
      rest = (0, _objectWithoutProperties2.default)(_props$composedModal, _excluded);

  var _props$modalBody = props.modalBody(),
      hasScrollingContent = _props$modalBody.hasScrollingContent;

  return /*#__PURE__*/_react.default.createElement(_ComposedModal.default, (0, _extends2.default)({}, rest, {
    size: size || undefined
  }), /*#__PURE__*/_react.default.createElement(_ModalHeader.ModalHeader, props.modalHeader()), /*#__PURE__*/_react.default.createElement(_ComposedModal.ModalBody, (0, _extends2.default)({}, props.modalBody(), {
    "aria-label": hasScrollingContent ? 'Modal content' : undefined
  }), /*#__PURE__*/_react.default.createElement("p", {
    style: {
      marginBottom: '1rem'
    }
  }, "Custom domains direct requests for your apps in this Cloud Foundry organization to a URL that you own. A custom domain can be a shared domain, a shared subdomain, or a shared domain and host."), /*#__PURE__*/_react.default.createElement(_TextInput.default, {
    "data-modal-primary-focus": true,
    id: "text-input-1",
    labelText: "Domain name",
    placeholder: "e.g. github.com",
    style: {
      marginBottom: '1rem'
    }
  }), /*#__PURE__*/_react.default.createElement(_Select.default, {
    id: "select-1",
    defaultValue: "us-south",
    labelText: "Region"
  }, /*#__PURE__*/_react.default.createElement(_SelectItem.default, {
    value: "us-south",
    text: "US South"
  }), /*#__PURE__*/_react.default.createElement(_SelectItem.default, {
    value: "us-east",
    text: "US East"
  })), /*#__PURE__*/_react.default.createElement("br", null), hasScrollingContent && scrollingContent), numberOfButtons > 0 && /*#__PURE__*/_react.default.createElement(_ModalFooter.ModalFooter, props.modalFooter(numberOfButtons)));
};

exports.Playground = Playground;

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_ComposedModal.default, {
    open: true
  }, /*#__PURE__*/_react.default.createElement(_ModalHeader.ModalHeader, {
    label: "Account resources",
    title: "Add a custom domain"
  }), /*#__PURE__*/_react.default.createElement(_ComposedModal.ModalBody, null, /*#__PURE__*/_react.default.createElement("p", {
    style: {
      marginBottom: '1rem'
    }
  }, "Custom domains direct requests for your apps in this Cloud Foundry organization to a URL that you own. A custom domain can be a shared domain, a shared subdomain, or a shared domain and host."), /*#__PURE__*/_react.default.createElement(_TextInput.default, {
    "data-modal-primary-focus": true,
    id: "text-input-1",
    labelText: "Domain name",
    placeholder: "e.g. github.com",
    style: {
      marginBottom: '1rem'
    }
  }), /*#__PURE__*/_react.default.createElement(_Select.default, {
    id: "select-1",
    defaultValue: "us-south",
    labelText: "Region"
  }, /*#__PURE__*/_react.default.createElement(_SelectItem.default, {
    value: "us-south",
    text: "US South"
  }), /*#__PURE__*/_react.default.createElement(_SelectItem.default, {
    value: "us-east",
    text: "US East"
  }))), /*#__PURE__*/_react.default.createElement(_ModalFooter.ModalFooter, {
    primaryButtonText: "Add",
    secondaryButtonText: "Cancel"
  }));
};

exports.Default = Default;
Default.story = {
  name: 'Composed Modal'
};

var PassiveModal = function PassiveModal() {
  return /*#__PURE__*/_react.default.createElement(_ComposedModal.default, {
    open: true
  }, /*#__PURE__*/_react.default.createElement(_ModalHeader.ModalHeader, {
    title: "You have been successfully signed out"
  }), /*#__PURE__*/_react.default.createElement(_ComposedModal.ModalBody, null));
};

exports.PassiveModal = PassiveModal;

var WithStateManager = function WithStateManager() {
  /**
   * Simple state manager for modals.
   */
  var ModalStateManager = function ModalStateManager(_ref2) {
    var LauncherContent = _ref2.renderLauncher,
        ModalContent = _ref2.children;

    var _useState = (0, _react.useState)(false),
        _useState2 = (0, _slicedToArray2.default)(_useState, 2),
        open = _useState2[0],
        setOpen = _useState2[1];

    return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, !ModalContent || typeof document === 'undefined' ? null : /*#__PURE__*/_reactDom.default.createPortal( /*#__PURE__*/_react.default.createElement(ModalContent, {
      open: open,
      setOpen: setOpen
    }), document.body), LauncherContent && /*#__PURE__*/_react.default.createElement(LauncherContent, {
      open: open,
      setOpen: setOpen
    }));
  };

  return /*#__PURE__*/_react.default.createElement(ModalStateManager, {
    renderLauncher: function renderLauncher(_ref3) {
      var setOpen = _ref3.setOpen;
      return /*#__PURE__*/_react.default.createElement(_Button.default, {
        onClick: function onClick() {
          return setOpen(true);
        }
      }, "Launch composed modal");
    }
  }, function (_ref4) {
    var open = _ref4.open,
        setOpen = _ref4.setOpen;
    return /*#__PURE__*/_react.default.createElement(_ComposedModal.default, {
      open: open,
      onClose: function onClose() {
        return setOpen(false);
      }
    }, /*#__PURE__*/_react.default.createElement(_ModalHeader.ModalHeader, {
      label: "Account resources",
      title: "Add a custom domain"
    }), /*#__PURE__*/_react.default.createElement(_ComposedModal.ModalBody, null, /*#__PURE__*/_react.default.createElement("p", {
      style: {
        marginBottom: '1rem'
      }
    }, "Custom domains direct requests for your apps in this Cloud Foundry organization to a URL that you own. A custom domain can be a shared domain, a shared subdomain, or a shared domain and host."), /*#__PURE__*/_react.default.createElement(_TextInput.default, {
      "data-modal-primary-focus": true,
      id: "text-input-1",
      labelText: "Domain name",
      placeholder: "e.g. github.com",
      style: {
        marginBottom: '1rem'
      }
    }), /*#__PURE__*/_react.default.createElement(_Select.default, {
      id: "select-1",
      defaultValue: "us-south",
      labelText: "Region"
    }, /*#__PURE__*/_react.default.createElement(_SelectItem.default, {
      value: "us-south",
      text: "US South"
    }), /*#__PURE__*/_react.default.createElement(_SelectItem.default, {
      value: "us-east",
      text: "US East"
    }))), /*#__PURE__*/_react.default.createElement(_ModalFooter.ModalFooter, {
      primaryButtonText: "Add",
      secondaryButtonText: "Cancel"
    }));
  });
};

exports.WithStateManager = WithStateManager;
