// carbon-components-react/lib/components/ContentSwitcher/next/ContentSwitcher-story.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require("./");

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

/* eslint-disable storybook/story-exports */
var _default = {
  title: 'Experimental/unstable_ContentSwitcher',
  component: _.ContentSwitcher,
  subcomponents: {
    ContentTabs: _.ContentTabs,
    ContentTab: _.ContentTab,
    ContentPanels: _.ContentPanels,
    ContentPanel: _.ContentPanel
  },
  includeStories: []
};
exports.default = _default;

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_.ContentSwitcher, null, /*#__PURE__*/_react.default.createElement(_.ContentTabs, {
    activation: "manual",
    "aria-label": "Example content switcher"
  }, /*#__PURE__*/_react.default.createElement(_.ContentTab, null, "Tab 1"), /*#__PURE__*/_react.default.createElement(_.ContentTab, null, "Tab 2"), /*#__PURE__*/_react.default.createElement(_.ContentTab, null, "Tab 3")), /*#__PURE__*/_react.default.createElement(_.ContentPanels, null, /*#__PURE__*/_react.default.createElement(_.ContentPanel, null, "Panel 1"), /*#__PURE__*/_react.default.createElement(_.ContentPanel, null, "Panel 2"), /*#__PURE__*/_react.default.createElement(_.ContentPanel, null, "Panel 3")));
};

exports.Default = Default;
