// carbon-components-react/lib/components/RadioButton/next/RadioButton.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Default = void 0;

var _ = _interopRequireDefault(require("../"));

var _RadioButtonGroup = _interopRequireDefault(require("../../RadioButtonGroup"));

var _react = _interopRequireDefault(require("react"));

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
var _default = {
  title: 'Components/RadioButton',
  component: _.default,
  subcomponents: {
    RadioButtonGroup: _RadioButtonGroup.default
  }
};
exports.default = _default;

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_RadioButtonGroup.default, {
    legendText: "Radio button heading",
    name: "radio-button-group",
    defaultSelected: "radio-1"
  }, /*#__PURE__*/_react.default.createElement(_.default, {
    labelText: "Option 1",
    value: "radio-1",
    id: "radio-1",
    disabled: true
  }), /*#__PURE__*/_react.default.createElement(_.default, {
    labelText: "Option 2",
    value: "radio-2",
    id: "radio-2"
  }), /*#__PURE__*/_react.default.createElement(_.default, {
    labelText: "Option 3",
    value: "radio-3",
    id: "radio-3"
  }));
};

exports.Default = Default;
