// carbon-components-react/lib/components/Grid/next/Grid.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Subgrid = exports.Responsive = exports.Offset = exports.Narrow = exports.MixedGridModes = exports.GridStartEnd = exports.FullWidth = exports.Default = exports.Condensed = void 0;

require("./Grid.stories.scss");

var _react = _interopRequireDefault(require("react"));

var _Grid = require("../../Grid");

var _Grid2 = _interopRequireDefault(require("./Grid.mdx"));

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
var _default = {
  title: 'Elements/Grid',
  component: _Grid.Grid,
  subcomponents: {
    Column: _Grid.Column
  },
  parameters: {
    controls: {
      hideNoControlsWarning: true
    },
    docs: {
      page: _Grid2.default
    }
  },
  decorators: [function (Story) {
    return /*#__PURE__*/_react.default.createElement("div", {
      className: "sb-css-grid-container"
    }, /*#__PURE__*/_react.default.createElement(Story, null));
  }]
};
exports.default = _default;

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_Grid.Grid, null, /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 4
  }), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 4
  }), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 4
  }), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 4
  }));
};

exports.Default = Default;

var Narrow = function Narrow() {
  return /*#__PURE__*/_react.default.createElement(_Grid.Grid, {
    narrow: true
  }, /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 4
  }), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 4
  }), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 4
  }), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 4
  }));
};

exports.Narrow = Narrow;

var Condensed = function Condensed() {
  return /*#__PURE__*/_react.default.createElement(_Grid.Grid, {
    condensed: true
  }, /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 4
  }), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 4
  }), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 4
  }), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 4
  }));
};

exports.Condensed = Condensed;

var FullWidth = function FullWidth() {
  return /*#__PURE__*/_react.default.createElement(_Grid.Grid, {
    fullWidth: true
  }, /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 4
  }), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 4
  }), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 4
  }), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 4
  }));
};

exports.FullWidth = FullWidth;

var Responsive = function Responsive() {
  return /*#__PURE__*/_react.default.createElement(_Grid.Grid, null, /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 2,
    md: 4,
    lg: 6
  }, /*#__PURE__*/_react.default.createElement("p", null, "Small: Span 2 of 4"), /*#__PURE__*/_react.default.createElement("p", null, "Medium: Span 4 of 8"), /*#__PURE__*/_react.default.createElement("p", null, "Large: Span 6 of 16")), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 2,
    md: 2,
    lg: 3
  }, /*#__PURE__*/_react.default.createElement("p", null, "Small: Span 2 of 4"), /*#__PURE__*/_react.default.createElement("p", null, "Medium: Span 2 of 8"), /*#__PURE__*/_react.default.createElement("p", null, "Large: Span 3 of 16")), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 0,
    md: 2,
    lg: 3
  }, /*#__PURE__*/_react.default.createElement("p", null, "Small: Span 0 of 4"), /*#__PURE__*/_react.default.createElement("p", null, "Medium: Span 2 of 8"), /*#__PURE__*/_react.default.createElement("p", null, "Large: Span 3 of 16")), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 0,
    md: 0,
    lg: 4
  }, /*#__PURE__*/_react.default.createElement("p", null, "Small: Span 0 of 4"), /*#__PURE__*/_react.default.createElement("p", null, "Medium: Span 0 of 8"), /*#__PURE__*/_react.default.createElement("p", null, "Large: Span 4 of 16")), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: "25%",
    md: "50%",
    lg: "75%"
  }, /*#__PURE__*/_react.default.createElement("p", null, "Small: Span 25%"), /*#__PURE__*/_react.default.createElement("p", null, "Medium: Span 50%"), /*#__PURE__*/_react.default.createElement("p", null, "Large: Span 75%")));
};

exports.Responsive = Responsive;

var Subgrid = function Subgrid() {
  return /*#__PURE__*/_react.default.createElement(_Grid.Grid, null, /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 2,
    md: 4,
    lg: 3
  }, /*#__PURE__*/_react.default.createElement("p", null, "Small: Span 2 of 4"), /*#__PURE__*/_react.default.createElement("p", null, "Medium: Span 4 of 8"), /*#__PURE__*/_react.default.createElement("p", null, "Large: Span 3 of 16")), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 2,
    md: 4,
    lg: 10
  }, /*#__PURE__*/_react.default.createElement("p", null, "Small: Span 2 of 4"), /*#__PURE__*/_react.default.createElement("p", null, "Medium: Span 4 of 8"), /*#__PURE__*/_react.default.createElement("p", null, "Large: Span 10 of 16"), /*#__PURE__*/_react.default.createElement(_Grid.Grid, {
    className: "example"
  }, /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 1,
    md: 1,
    lg: 2
  }, /*#__PURE__*/_react.default.createElement("p", null, "sm=", 1), " ", /*#__PURE__*/_react.default.createElement("p", null, "md=", 1), " ", /*#__PURE__*/_react.default.createElement("p", null, "lg=", 2)), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 1,
    md: 1,
    lg: 2
  }, /*#__PURE__*/_react.default.createElement("p", null, "sm=", 1), " ", /*#__PURE__*/_react.default.createElement("p", null, "md=", 1), " ", /*#__PURE__*/_react.default.createElement("p", null, "lg=", 2)), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 0,
    md: 1,
    lg: 1
  }, /*#__PURE__*/_react.default.createElement("p", null, "sm=", 0), " ", /*#__PURE__*/_react.default.createElement("p", null, "md=", 1), " ", /*#__PURE__*/_react.default.createElement("p", null, "lg=", 1)), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 0,
    md: 1,
    lg: 1
  }, /*#__PURE__*/_react.default.createElement("p", null, "sm=", 0), " ", /*#__PURE__*/_react.default.createElement("p", null, "md=", 1), " ", /*#__PURE__*/_react.default.createElement("p", null, "lg=", 1)), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 0,
    md: 0,
    lg: 1
  }, /*#__PURE__*/_react.default.createElement("p", null, "sm=", 0), " ", /*#__PURE__*/_react.default.createElement("p", null, "md=", 0), " ", /*#__PURE__*/_react.default.createElement("p", null, "lg=", 1)), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 0,
    md: 0,
    lg: 1
  }, /*#__PURE__*/_react.default.createElement("p", null, "sm=", 0), " ", /*#__PURE__*/_react.default.createElement("p", null, "md=", 0), " ", /*#__PURE__*/_react.default.createElement("p", null, "lg=", 1)), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 0,
    md: 0,
    lg: 1
  }, /*#__PURE__*/_react.default.createElement("p", null, "sm=", 0), " ", /*#__PURE__*/_react.default.createElement("p", null, "md=", 0), " ", /*#__PURE__*/_react.default.createElement("p", null, "lg=", 1)), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 0,
    md: 0,
    lg: 1
  }, /*#__PURE__*/_react.default.createElement("p", null, "sm=", 0), " ", /*#__PURE__*/_react.default.createElement("p", null, "md=", 0), " ", /*#__PURE__*/_react.default.createElement("p", null, "lg=", 1)))), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 0,
    md: 0,
    lg: 3
  }, /*#__PURE__*/_react.default.createElement("p", null, "Small: Span 0 of 4"), /*#__PURE__*/_react.default.createElement("p", null, "Medium: Span 0 of 8"), /*#__PURE__*/_react.default.createElement("p", null, "Large: Span 3 of 16")));
};

exports.Subgrid = Subgrid;

var MixedGridModes = function MixedGridModes() {
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_Grid.Grid, null, /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    span: 8
  }, /*#__PURE__*/_react.default.createElement(_Grid.Grid, null, /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    span: 8
  }, /*#__PURE__*/_react.default.createElement(_Grid.Grid, {
    narrow: true
  }, /*#__PURE__*/_react.default.createElement(_Grid.Column, null, /*#__PURE__*/_react.default.createElement(_Grid.ColumnHang, null, "Text")), /*#__PURE__*/_react.default.createElement(_Grid.Column, null, /*#__PURE__*/_react.default.createElement(_Grid.ColumnHang, null, "Text")), /*#__PURE__*/_react.default.createElement(_Grid.Column, null, /*#__PURE__*/_react.default.createElement(_Grid.ColumnHang, null, "Text")), /*#__PURE__*/_react.default.createElement(_Grid.Column, null, /*#__PURE__*/_react.default.createElement(_Grid.ColumnHang, null, "Text")), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    span: 4
  }, /*#__PURE__*/_react.default.createElement(_Grid.Grid, null, /*#__PURE__*/_react.default.createElement(_Grid.Column, null, "Text"), /*#__PURE__*/_react.default.createElement(_Grid.Column, null, "Text"), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    span: 2
  }, /*#__PURE__*/_react.default.createElement(_Grid.Grid, {
    condensed: true
  }, /*#__PURE__*/_react.default.createElement(_Grid.Column, null, /*#__PURE__*/_react.default.createElement(_Grid.ColumnHang, null, "Text")), /*#__PURE__*/_react.default.createElement(_Grid.Column, null, /*#__PURE__*/_react.default.createElement(_Grid.ColumnHang, null, "Text"))))))))))), /*#__PURE__*/_react.default.createElement(_Grid.Grid, {
    narrow: true
  }, /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    span: 8
  }, /*#__PURE__*/_react.default.createElement(_Grid.Grid, null, /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    span: 4
  }), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    span: 4
  }, /*#__PURE__*/_react.default.createElement(_Grid.Grid, {
    narrow: true
  }, /*#__PURE__*/_react.default.createElement(_Grid.Column, null, /*#__PURE__*/_react.default.createElement(_Grid.ColumnHang, null, "Text")), /*#__PURE__*/_react.default.createElement(_Grid.Column, null, /*#__PURE__*/_react.default.createElement(_Grid.ColumnHang, null, "Text")), /*#__PURE__*/_react.default.createElement(_Grid.Column, null, /*#__PURE__*/_react.default.createElement(_Grid.ColumnHang, null, "Text")), /*#__PURE__*/_react.default.createElement(_Grid.Column, null, /*#__PURE__*/_react.default.createElement(_Grid.ColumnHang, null, "Text"))))))));
};

exports.MixedGridModes = MixedGridModes;

var GridStartEnd = function GridStartEnd() {
  return /*#__PURE__*/_react.default.createElement(_Grid.Grid, null, /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: {
      span: 1,
      start: 4
    },
    md: {
      span: 2,
      start: 7
    },
    lg: {
      span: 4,
      start: 13
    }
  }, "span, start"), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: {
      span: 2,
      end: 5
    },
    md: {
      span: 4,
      end: 9
    },
    lg: {
      span: 8,
      end: 17
    }
  }, "span, end"), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: {
      start: 1,
      end: 4
    },
    md: {
      start: 3,
      end: 9
    },
    lg: {
      start: 5,
      end: 17
    }
  }, "start, end"));
};

exports.GridStartEnd = GridStartEnd;

var Offset = function Offset() {
  return /*#__PURE__*/_react.default.createElement(_Grid.Grid, null, /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: {
      span: 1,
      offset: 3
    },
    md: {
      span: 2,
      offset: 6
    },
    lg: {
      span: 4,
      offset: 12
    }
  }), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: {
      span: 2,
      offset: 2
    },
    md: {
      span: 4,
      offset: 4
    },
    lg: {
      span: 8,
      offset: 8
    }
  }), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: {
      span: 3,
      offset: 1
    },
    md: {
      span: 6,
      offset: 2
    },
    lg: {
      span: 12,
      offset: 4
    }
  }), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: {
      span: 4
    },
    md: {
      span: 8
    },
    lg: {
      span: 16
    }
  }), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: {
      span: '25%',
      offset: 1
    },
    md: {
      span: '50%',
      offset: 2
    },
    lg: {
      span: '75%',
      offset: 4
    }
  }));
};

exports.Offset = Offset;
