// carbon-components-react/lib/components/UIShell/HeaderPanel.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _usePrefix = require("../../internal/usePrefix");

var _excluded = ["children", "className", "expanded"];

var HeaderPanel = /*#__PURE__*/_react.default.forwardRef(function HeaderPanel(_ref, ref) {
  var _cx;

  var children = _ref.children,
      customClassName = _ref.className,
      expanded = _ref.expanded,
      other = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  var prefix = (0, _usePrefix.usePrefix)();
  var className = (0, _classnames.default)("".concat(prefix, "--header-panel"), (_cx = {}, (0, _defineProperty2.default)(_cx, "".concat(prefix, "--header-panel--expanded"), expanded), (0, _defineProperty2.default)(_cx, customClassName, !!customClassName), _cx));
  return /*#__PURE__*/_react.default.createElement("div", (0, _extends2.default)({}, other, {
    className: className,
    ref: ref
  }), children);
});

HeaderPanel.propTypes = {
  /**
   * The content that will render inside of the `HeaderPanel`
   */
  children: _propTypes.default.node,

  /**
   * Optionally provide a custom class to apply to the underlying `<li>` node
   */
  className: _propTypes.default.string,

  /**
   * Specify whether the panel is expanded
   */
  expanded: _propTypes.default.bool
};
HeaderPanel.displayName = 'HeaderPanel';
var _default = HeaderPanel;
exports.default = _default;
