// carbon-components-react/lib/components/ToggleSmall/ToggleSmall.Skeleton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _defineProperty4 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _usePrefix = require("../../internal/usePrefix");

var _defineProperty2;

var _excluded = ["id", "labelText", "className"];

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

var ToggleSmallSkeleton = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(ToggleSmallSkeleton, _React$Component);

  var _super = _createSuper(ToggleSmallSkeleton);

  function ToggleSmallSkeleton() {
    (0, _classCallCheck2.default)(this, ToggleSmallSkeleton);
    return _super.apply(this, arguments);
  }

  (0, _createClass2.default)(ToggleSmallSkeleton, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          id = _this$props.id,
          labelText = _this$props.labelText,
          className = _this$props.className,
          rest = (0, _objectWithoutProperties2.default)(_this$props, _excluded);
      return /*#__PURE__*/_react.default.createElement(_usePrefix.PrefixContext.Consumer, null, function (prefix) {
        return /*#__PURE__*/_react.default.createElement("div", (0, _extends2.default)({
          className: (0, _classnames.default)("".concat(prefix, "--form-item"), className)
        }, rest), /*#__PURE__*/_react.default.createElement("input", {
          type: "checkbox",
          id: id,
          className: "".concat(prefix, "--toggle ").concat(prefix, "--toggle--small ").concat(prefix, "--skeleton")
        }), /*#__PURE__*/_react.default.createElement("label", {
          className: "".concat(prefix, "--toggle__label ").concat(prefix, "--skeleton"),
          htmlFor: id
        }, labelText && /*#__PURE__*/_react.default.createElement("span", {
          className: "".concat(prefix, "--toggle__label-text")
        }, labelText), /*#__PURE__*/_react.default.createElement("span", {
          className: "".concat(prefix, "--toggle__appearance")
        }, /*#__PURE__*/_react.default.createElement("svg", {
          className: "".concat(prefix, "--toggle__check"),
          width: "6px",
          height: "5px",
          viewBox: "0 0 6 5"
        }, /*#__PURE__*/_react.default.createElement("path", {
          d: "M2.2403 2.7299L4.9245 0 6 1.1117 2.2384 5 0 2.6863 1.0612 1.511z"
        })))));
      });
    }
  }]);
  return ToggleSmallSkeleton;
}(_react.default.Component);

exports.default = ToggleSmallSkeleton;
(0, _defineProperty4.default)(ToggleSmallSkeleton, "propTypes", (_defineProperty2 = {}, (0, _defineProperty4.default)(_defineProperty2, 'aria-label', _propTypes.default.string.isRequired), (0, _defineProperty4.default)(_defineProperty2, "className", _propTypes.default.string), (0, _defineProperty4.default)(_defineProperty2, "id", _propTypes.default.string), (0, _defineProperty4.default)(_defineProperty2, "labelText", _propTypes.default.string), _defineProperty2));
(0, _defineProperty4.default)(ToggleSmallSkeleton, "defaultProps", (0, _defineProperty4.default)({}, 'aria-label', 'Toggle is loading'));
