// carbon-components-react/lib/components/StructuredList/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StructuredListRow = exports.StructuredListInput = exports.StructuredListHead = exports.StructuredListCell = exports.StructuredListBody = void 0;
Object.defineProperty(exports, "StructuredListSkeleton", {
  enumerable: true,
  get: function get() {
    return _StructuredList3.default;
  }
});
exports.StructuredListWrapper = void 0;

var _StructuredList = require("./next/StructuredList");

var _StructuredList2 = require("./StructuredList");

var _ComponentToggle = require("../../internal/ComponentToggle");

var _StructuredList3 = _interopRequireDefault(require("./StructuredList.Skeleton"));

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
var StructuredListWrapper = (0, _ComponentToggle.createComponentToggle)({
  name: 'StructuredListWrapper',
  next: _StructuredList.StructuredListWrapper,
  classic: _StructuredList2.StructuredListWrapper
});
exports.StructuredListWrapper = StructuredListWrapper;
var StructuredListHead = (0, _ComponentToggle.createComponentToggle)({
  name: 'StructuredListHead',
  next: _StructuredList.StructuredListHead,
  classic: _StructuredList2.StructuredListHead
});
exports.StructuredListHead = StructuredListHead;
var StructuredListInput = (0, _ComponentToggle.createComponentToggle)({
  name: 'StructuredListInput',
  next: _StructuredList.StructuredListInput,
  classic: _StructuredList2.StructuredListInput
});
exports.StructuredListInput = StructuredListInput;
var StructuredListBody = (0, _ComponentToggle.createComponentToggle)({
  name: 'StructuredListBody',
  next: _StructuredList.StructuredListBody,
  classic: _StructuredList2.StructuredListBody
});
exports.StructuredListBody = StructuredListBody;
var StructuredListRow = (0, _ComponentToggle.createComponentToggle)({
  name: 'StructuredListRow',
  next: _StructuredList.StructuredListRow,
  classic: _StructuredList2.StructuredListRow
});
exports.StructuredListRow = StructuredListRow;
var StructuredListCell = (0, _ComponentToggle.createComponentToggle)({
  name: 'StructuredListCell',
  next: _StructuredList.StructuredListCell,
  classic: _StructuredList2.StructuredListCell
});
exports.StructuredListCell = StructuredListCell;
