// carbon-components-react/lib/components/Toggletip/next/Toggletip.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Playground = exports.Default = void 0;

var _iconsReact = require("@carbon/icons-react");

var _react = _interopRequireDefault(require("react"));

var _Button = _interopRequireDefault(require("../../Button"));

var _Link = _interopRequireDefault(require("../../Link"));

var _Toggletip = require("../../Toggletip");

var _Toggletip2 = _interopRequireDefault(require("./Toggletip.mdx"));

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
var _default = {
  title: 'Components/Toggletip',
  component: _Toggletip.Toggletip,
  subcomponents: {
    ToggletipLabel: _Toggletip.ToggletipLabel,
    ToggletipButton: _Toggletip.ToggletipButton,
    ToggletipContent: _Toggletip.ToggletipContent,
    ToggletipActions: _Toggletip.ToggletipActions
  },
  argTypes: {
    as: {
      table: {
        disable: true
      }
    },
    children: {
      table: {
        disable: true
      }
    },
    className: {
      table: {
        disable: true
      }
    }
  },
  parameters: {
    docs: {
      page: _Toggletip2.default
    }
  }
};
exports.default = _default;

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      display: 'flex',
      justifyContent: 'center'
    }
  }, /*#__PURE__*/_react.default.createElement(_Toggletip.ToggletipLabel, null, "Toggletip label"), /*#__PURE__*/_react.default.createElement(_Toggletip.Toggletip, {
    defaultOpen: true
  }, /*#__PURE__*/_react.default.createElement(_Toggletip.ToggletipButton, {
    label: "Show information"
  }, /*#__PURE__*/_react.default.createElement(_iconsReact.Information16, null)), /*#__PURE__*/_react.default.createElement(_Toggletip.ToggletipContent, null, /*#__PURE__*/_react.default.createElement("p", null, "Lorem ipsum dolor sit amet, di os consectetur adipiscing elit, sed do eiusmod tempor incididunt ut fsil labore et dolore magna aliqua."), /*#__PURE__*/_react.default.createElement(_Toggletip.ToggletipActions, null, /*#__PURE__*/_react.default.createElement(_Link.default, {
    href: "#"
  }, "Link action"), /*#__PURE__*/_react.default.createElement(_Button.default, {
    size: "sm"
  }, "Button")))));
};

exports.Default = Default;

var PlaygroundStory = function PlaygroundStory(controls) {
  var align = controls.align;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_Toggletip.ToggletipLabel, null, "Toggletip label"), /*#__PURE__*/_react.default.createElement(_Toggletip.Toggletip, {
    align: align,
    defaultOpen: true
  }, /*#__PURE__*/_react.default.createElement(_Toggletip.ToggletipButton, {
    label: "Show information"
  }, /*#__PURE__*/_react.default.createElement(_iconsReact.Information16, null)), /*#__PURE__*/_react.default.createElement(_Toggletip.ToggletipContent, null, /*#__PURE__*/_react.default.createElement("p", null, "Lorem ipsum dolor sit amet, di os consectetur adipiscing elit, sed do eiusmod tempor incididunt ut fsil labore et dolore magna aliqua."), /*#__PURE__*/_react.default.createElement(_Toggletip.ToggletipActions, null, /*#__PURE__*/_react.default.createElement(_Link.default, {
    href: "#"
  }, "Link action"), /*#__PURE__*/_react.default.createElement(_Button.default, {
    size: "sm"
  }, "Button")))));
};

var Playground = PlaygroundStory.bind({});
exports.Playground = Playground;
Playground.argTypes = {
  align: {
    options: ['top', 'bottom', 'left', 'right'],
    control: {
      type: 'select'
    }
  }
};
Playground.story = {
  decorators: [function (story) {
    return /*#__PURE__*/_react.default.createElement("div", {
      style: {
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center'
      }
    }, story());
  }]
};
