// carbon-components-react/lib/components/Grid/Grid-story.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.ResponsiveGrid = exports.Offset = exports.NarrowColumns = exports.Narrow = exports.MixedGridModes = exports.FullWidth = exports.ExperimentalCSSGrid = exports.CondensedColumns = exports.Condensed = exports.AutoColumns = void 0;

require("./Grid-story.scss");

var _react = _interopRequireDefault(require("react"));

var _ = require("./");

var _Grid = _interopRequireDefault(require("./Grid.mdx"));

var _FeatureFlags = require("../FeatureFlags");

var _Heading = require("../Heading");

var _default = {
  title: 'Components/Grid',
  component: _.Grid,
  subcomponents: {
    Row: _.Row,
    Column: _.Column
  },
  decorators: [function (storyFn) {
    return /*#__PURE__*/_react.default.createElement("div", {
      id: "templates"
    }, storyFn());
  }],
  parameters: {
    docs: {
      page: _Grid.default
    }
  }
};
exports.default = _default;

function DemoContent(_ref) {
  var children = _ref.children;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "outside"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "inside"
  }, children));
}

var ExperimentalCSSGrid = function ExperimentalCSSGrid() {
  return /*#__PURE__*/_react.default.createElement(_FeatureFlags.FeatureFlags, {
    flags: {
      'enable-css-grid': true
    }
  }, /*#__PURE__*/_react.default.createElement(_Heading.Heading, null, "Wide"), /*#__PURE__*/_react.default.createElement(_.Grid, null, /*#__PURE__*/_react.default.createElement(_.Column, {
    sm: 1,
    md: 2,
    lg: 4
  }, "Column"), /*#__PURE__*/_react.default.createElement(_.Column, {
    sm: 1,
    md: 2,
    lg: 4
  }, "Column"), /*#__PURE__*/_react.default.createElement(_.Column, {
    sm: 1,
    md: 2,
    lg: 4
  }, "Column"), /*#__PURE__*/_react.default.createElement(_.Column, {
    sm: 1,
    md: 2,
    lg: 4
  }, "Column"), /*#__PURE__*/_react.default.createElement(_.Grid, null, /*#__PURE__*/_react.default.createElement(_.Column, {
    sm: 2,
    md: 4,
    lg: 8
  }, "Subgrid"), /*#__PURE__*/_react.default.createElement(_.Column, {
    sm: 2,
    md: 4,
    lg: 8
  }, "Subgrid"))), /*#__PURE__*/_react.default.createElement(_Heading.Heading, null, "Condensed"), /*#__PURE__*/_react.default.createElement(_.Grid, {
    condensed: true
  }, /*#__PURE__*/_react.default.createElement(_.Column, {
    sm: 1,
    md: 2,
    lg: 4
  }, "Column"), /*#__PURE__*/_react.default.createElement(_.Column, {
    sm: 1,
    md: 2,
    lg: 4
  }, "Column"), /*#__PURE__*/_react.default.createElement(_.Column, {
    sm: 1,
    md: 2,
    lg: 4
  }, "Column"), /*#__PURE__*/_react.default.createElement(_.Column, {
    sm: 1,
    md: 2,
    lg: 4
  }, "Column")));
};

exports.ExperimentalCSSGrid = ExperimentalCSSGrid;

var AutoColumns = function AutoColumns() {
  return /*#__PURE__*/_react.default.createElement(_.Grid, null, /*#__PURE__*/_react.default.createElement(_.Row, null, /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "Span 25%")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "Span 25%")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "Span 25%")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "Span 25%"))));
};

exports.AutoColumns = AutoColumns;

var ResponsiveGrid = function ResponsiveGrid() {
  return /*#__PURE__*/_react.default.createElement(_.Grid, null, /*#__PURE__*/_react.default.createElement(_.Row, null, /*#__PURE__*/_react.default.createElement(_.Column, {
    sm: 2,
    md: 4,
    lg: 6
  }, /*#__PURE__*/_react.default.createElement(DemoContent, null, /*#__PURE__*/_react.default.createElement("p", null, "Small: Span 2 of 4"), /*#__PURE__*/_react.default.createElement("p", null, "Medium: Span 4 of 8"), /*#__PURE__*/_react.default.createElement("p", null, "Large: Span 6 of 12"))), /*#__PURE__*/_react.default.createElement(_.Column, {
    sm: 2,
    md: 2,
    lg: 3
  }, /*#__PURE__*/_react.default.createElement(DemoContent, null, /*#__PURE__*/_react.default.createElement("p", null, "Small: Span 2 of 4"), /*#__PURE__*/_react.default.createElement("p", null, "Medium: Span 2 of 8"), /*#__PURE__*/_react.default.createElement("p", null, "Large: Span 3 of 12"))), /*#__PURE__*/_react.default.createElement(_.Column, {
    sm: 0,
    md: 2,
    lg: 3
  }, /*#__PURE__*/_react.default.createElement(DemoContent, null, /*#__PURE__*/_react.default.createElement("p", null, "Small: Span 0 of 4"), /*#__PURE__*/_react.default.createElement("p", null, "Medium: Span 2 of 8"), /*#__PURE__*/_react.default.createElement("p", null, "Large: Span 3 of 12")))));
};

exports.ResponsiveGrid = ResponsiveGrid;

var Offset = function Offset() {
  return /*#__PURE__*/_react.default.createElement(_.Grid, null, /*#__PURE__*/_react.default.createElement(_.Row, null, /*#__PURE__*/_react.default.createElement(_.Column, {
    sm: {
      span: 1,
      offset: 3
    }
  }, /*#__PURE__*/_react.default.createElement(DemoContent, null, "Small: offset 3")), /*#__PURE__*/_react.default.createElement(_.Column, {
    sm: {
      span: 2,
      offset: 2
    }
  }, /*#__PURE__*/_react.default.createElement(DemoContent, null, "Small: offset 2")), /*#__PURE__*/_react.default.createElement(_.Column, {
    sm: {
      span: 3,
      offset: 1
    }
  }, /*#__PURE__*/_react.default.createElement(DemoContent, null, "Small: offset 1")), /*#__PURE__*/_react.default.createElement(_.Column, {
    sm: {
      span: 4,
      offset: 0
    }
  }, /*#__PURE__*/_react.default.createElement(DemoContent, null, "Small: offset 0"))));
};

exports.Offset = Offset;

var Condensed = function Condensed() {
  return /*#__PURE__*/_react.default.createElement(_.Grid, {
    condensed: true
  }, /*#__PURE__*/_react.default.createElement(_.Row, null, /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4"))));
};

exports.Condensed = Condensed;

var CondensedColumns = function CondensedColumns() {
  return /*#__PURE__*/_react.default.createElement(_.Grid, null, /*#__PURE__*/_react.default.createElement(_.Row, null, /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4"))), /*#__PURE__*/_react.default.createElement(_.Row, {
    condensed: true
  }, /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4"))), /*#__PURE__*/_react.default.createElement(_.Row, null, /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4"))));
};

exports.CondensedColumns = CondensedColumns;

var Narrow = function Narrow() {
  return /*#__PURE__*/_react.default.createElement(_.Grid, {
    narrow: true
  }, /*#__PURE__*/_react.default.createElement(_.Row, null, /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4"))));
};

exports.Narrow = Narrow;

var NarrowColumns = function NarrowColumns() {
  return /*#__PURE__*/_react.default.createElement(_.Grid, null, /*#__PURE__*/_react.default.createElement(_.Row, null, /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4"))), /*#__PURE__*/_react.default.createElement(_.Row, {
    narrow: true
  }, /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4"))), /*#__PURE__*/_react.default.createElement(_.Row, null, /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4"))));
};

exports.NarrowColumns = NarrowColumns;

var FullWidth = function FullWidth() {
  return /*#__PURE__*/_react.default.createElement(_.Grid, {
    fullWidth: true
  }, /*#__PURE__*/_react.default.createElement(_.Row, null, /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4"))));
};

exports.FullWidth = FullWidth;

var MixedGridModes = function MixedGridModes() {
  return /*#__PURE__*/_react.default.createElement(_.Grid, null, /*#__PURE__*/_react.default.createElement(_.Row, null, /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "Wide")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4"))), /*#__PURE__*/_react.default.createElement(_.Row, {
    narrow: true
  }, /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "Narrow")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4"))), /*#__PURE__*/_react.default.createElement(_.Row, {
    condensed: true
  }, /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "Condensed")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4")), /*#__PURE__*/_react.default.createElement(_.Column, null, /*#__PURE__*/_react.default.createElement(DemoContent, null, "1/4"))));
};

exports.MixedGridModes = MixedGridModes;
