// carbon-components-react/lib/components/AspectRatio/next/AspectRatio.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Playground = exports.AspectRatioStory = void 0;

require("../AspectRatio-story.scss");

var _react = _interopRequireDefault(require("react"));

var _Grid = require("../../Grid");

var _ = require("../");

var _AspectRatio = _interopRequireDefault(require("../AspectRatio.mdx"));

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
var _default = {
  title: 'Components/AspectRatio',
  component: _.AspectRatio,
  decorators: [function (Story) {
    return /*#__PURE__*/_react.default.createElement("div", {
      className: "aspect-ratio-story"
    }, /*#__PURE__*/_react.default.createElement(Story, null));
  }],
  parameters: {
    docs: {
      page: _AspectRatio.default
    }
  }
};
exports.default = _default;

var AspectRatioStory = function AspectRatioStory() {
  return /*#__PURE__*/_react.default.createElement(_Grid.Grid, null, /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 1,
    md: 2,
    lg: 4
  }, /*#__PURE__*/_react.default.createElement(_.AspectRatio, {
    ratio: "1x1"
  }, "Content")), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 1,
    md: 2,
    lg: 4
  }, /*#__PURE__*/_react.default.createElement(_.AspectRatio, {
    ratio: "1x1"
  }, "Content")), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 1,
    md: 2,
    lg: 4
  }, /*#__PURE__*/_react.default.createElement(_.AspectRatio, {
    ratio: "1x1"
  }, "Content")), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
    sm: 1,
    md: 2,
    lg: 4
  }, /*#__PURE__*/_react.default.createElement(_.AspectRatio, {
    ratio: "1x1"
  }, "Content")));
};

exports.AspectRatioStory = AspectRatioStory;
AspectRatioStory.storyName = 'AspectRatio';
var Playground = {
  argTypes: {
    as: {
      control: {
        type: null
      }
    },
    children: {
      control: {
        type: null
      }
    },
    className: {
      control: {
        type: null
      }
    },
    ratio: {
      control: {
        type: 'select'
      },
      defaultValue: '1x1',
      options: ['16x9', '9x16', '2x1', '1x2', '4x3', '3x4', '1x1']
    }
  },
  render: function render(_ref) {
    var ratio = _ref.ratio;
    return /*#__PURE__*/_react.default.createElement(_Grid.Grid, null, /*#__PURE__*/_react.default.createElement(_Grid.Column, {
      sm: 1,
      md: 2,
      lg: 4
    }, /*#__PURE__*/_react.default.createElement(_.AspectRatio, {
      ratio: ratio
    }, "Content")), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
      sm: 1,
      md: 2,
      lg: 4
    }, /*#__PURE__*/_react.default.createElement(_.AspectRatio, {
      ratio: ratio
    }, "Content")), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
      sm: 1,
      md: 2,
      lg: 4
    }, /*#__PURE__*/_react.default.createElement(_.AspectRatio, {
      ratio: ratio
    }, "Content")), /*#__PURE__*/_react.default.createElement(_Grid.Column, {
      sm: 1,
      md: 2,
      lg: 4
    }, /*#__PURE__*/_react.default.createElement(_.AspectRatio, {
      ratio: ratio
    }, "Content")));
  }
};
exports.Playground = Playground;
