// carbon-components-react/es/components/FormGroup/next/FormGroup.stories.js
import _extends from "@babel/runtime/helpers/extends";

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from 'react';
import FormGroup from '../FormGroup';
import TextInput from '../../TextInput';
import RadioButtonGroup from '../../RadioButtonGroup';
import RadioButton from '../../RadioButton';
import Button from '../../Button';
import { Stack } from '../../Stack';
export default {
  title: 'Components/FormGroup',
  component: FormGroup,
  argTypes: {
    disabled: {
      control: {
        type: 'boolean'
      },
      defaultValue: false
    },
    // TODO remove hasMargin in v11
    hasMargin: {
      table: {
        disable: true
      }
    },
    legendText: {
      control: {
        type: 'text'
      },
      defaultValue: 'FormGroup Legend'
    },
    legendId: {
      control: {
        type: 'text'
      },
      defaultValue: 'formgroup-legend-id'
    },
    children: {
      control: false
    }
  }
};
export var Default = function Default(args) {
  return /*#__PURE__*/React.createElement(FormGroup, _extends({
    style: {
      maxWidth: '400px'
    }
  }, args), /*#__PURE__*/React.createElement(Stack, {
    gap: 7
  }, /*#__PURE__*/React.createElement(TextInput, {
    id: "one",
    labelText: "First Name"
  }), /*#__PURE__*/React.createElement(TextInput, {
    id: "two",
    labelText: "Last Name"
  }), /*#__PURE__*/React.createElement(RadioButtonGroup, {
    legendText: "Radio button heading",
    name: "radio-button-group",
    defaultSelected: "radio-1"
  }, /*#__PURE__*/React.createElement(RadioButton, {
    labelText: "Option 1",
    value: "radio-1",
    id: "radio-1"
  }), /*#__PURE__*/React.createElement(RadioButton, {
    labelText: "Option 2",
    value: "radio-2",
    id: "radio-2"
  }), /*#__PURE__*/React.createElement(RadioButton, {
    labelText: "Option 3",
    value: "radio-3",
    id: "radio-3"
  })), /*#__PURE__*/React.createElement(Button, null, "Submit")));
};
