// carbon-components-react/lib/components/DatePickerInput/next/DatePickerInput.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _iconsReact = require("@carbon/icons-react");

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _usePrefix = require("../../../internal/usePrefix");

var _deprecate = _interopRequireDefault(require("../../../prop-types/deprecate"));

var _excluded = ["datePickerType", "disabled", "helperText", "hideLabel", "id", "invalid", "invalidText", "labelText", "onClick", "onChange", "pattern", "placeholder", "size", "type", "warn", "warnText"],
    _excluded2 = ["openCalendar", "iconDescription"],
    _excluded3 = ["datePickerType", "invalid", "warn"];

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

var DatePickerInput = /*#__PURE__*/_react.default.forwardRef(function DatePickerInput(props, ref) {
  var _cx, _cx2, _cx4;

  var datePickerType = props.datePickerType,
      _props$disabled = props.disabled,
      disabled = _props$disabled === void 0 ? false : _props$disabled,
      helperText = props.helperText,
      hideLabel = props.hideLabel,
      id = props.id,
      _props$invalid = props.invalid,
      invalid = _props$invalid === void 0 ? false : _props$invalid,
      invalidText = props.invalidText,
      labelText = props.labelText,
      _props$onClick = props.onClick,
      _onClick = _props$onClick === void 0 ? function () {} : _props$onClick,
      _props$onChange = props.onChange,
      _onChange = _props$onChange === void 0 ? function () {} : _props$onChange,
      _props$pattern = props.pattern,
      pattern = _props$pattern === void 0 ? '\\d{1,2}\\/\\d{1,2}\\/\\d{4}' : _props$pattern,
      placeholder = props.placeholder,
      _props$size = props.size,
      size = _props$size === void 0 ? 'md' : _props$size,
      _props$type = props.type,
      type = _props$type === void 0 ? 'text' : _props$type,
      warn = props.warn,
      warnText = props.warnText,
      rest = (0, _objectWithoutProperties2.default)(props, _excluded);

  var prefix = (0, _usePrefix.usePrefix)();
  var datePickerInputProps = {
    id: id,
    onChange: function onChange(event) {
      if (!disabled) {
        _onChange(event);
      }
    },
    onClick: function onClick(event) {
      if (!disabled) {
        _onClick(event);
      }
    },
    pattern: pattern,
    placeholder: placeholder,
    type: type
  };
  var wrapperClasses = (0, _classnames.default)("".concat(prefix, "--date-picker-input__wrapper"), (_cx = {}, (0, _defineProperty2.default)(_cx, "".concat(prefix, "--date-picker-input__wrapper--invalid"), invalid), (0, _defineProperty2.default)(_cx, "".concat(prefix, "--date-picker-input__wrapper--warn"), warn), _cx));
  var labelClasses = (0, _classnames.default)("".concat(prefix, "--label"), (_cx2 = {}, (0, _defineProperty2.default)(_cx2, "".concat(prefix, "--visually-hidden"), hideLabel), (0, _defineProperty2.default)(_cx2, "".concat(prefix, "--label--disabled"), disabled), _cx2));
  var helperTextClasses = (0, _classnames.default)("".concat(prefix, "--form__helper-text"), (0, _defineProperty2.default)({}, "".concat(prefix, "--form__helper-text--disabled"), disabled));
  var inputClasses = (0, _classnames.default)("".concat(prefix, "--date-picker__input"), (_cx4 = {}, (0, _defineProperty2.default)(_cx4, "".concat(prefix, "--date-picker__input--").concat(size), size), (0, _defineProperty2.default)(_cx4, "".concat(prefix, "--date-picker__input--invalid"), invalid), _cx4));
  var containerClasses = (0, _classnames.default)("".concat(prefix, "--date-picker-container"), (0, _defineProperty2.default)({}, "".concat(prefix, "--date-picker--nolabel"), !labelText));

  var _rest = _objectSpread({}, rest),
      openCalendar = _rest.openCalendar,
      iconDescription = _rest.iconDescription,
      other = (0, _objectWithoutProperties2.default)(_rest, _excluded2);

  var input = invalid ? /*#__PURE__*/_react.default.createElement("input", (0, _extends2.default)({}, other, datePickerInputProps, {
    disabled: disabled,
    ref: ref,
    "data-invalid": true,
    className: inputClasses
  })) : /*#__PURE__*/_react.default.createElement("input", (0, _extends2.default)({}, other, datePickerInputProps, {
    disabled: disabled,
    className: inputClasses,
    ref: ref
  }));
  return /*#__PURE__*/_react.default.createElement("div", {
    className: containerClasses
  }, labelText && /*#__PURE__*/_react.default.createElement("label", {
    htmlFor: id,
    className: labelClasses
  }, labelText), /*#__PURE__*/_react.default.createElement("div", {
    className: wrapperClasses
  }, input, /*#__PURE__*/_react.default.createElement(DatePickerIcon, {
    datePickerType: datePickerType,
    iconDescription: iconDescription,
    invalid: invalid,
    openCalendar: openCalendar,
    warn: warn
  })), invalid && /*#__PURE__*/_react.default.createElement("div", {
    className: "".concat(prefix, "--form-requirement")
  }, invalidText), warn && /*#__PURE__*/_react.default.createElement("div", {
    className: "".concat(prefix, "--form-requirement")
  }, warnText), helperText && /*#__PURE__*/_react.default.createElement("div", {
    className: helperTextClasses
  }, helperText));
});

DatePickerInput.propTypes = {
  /**
   * The type of the date picker:
   *
   * * `simple` - Without calendar dropdown.
   * * `single` - With calendar dropdown and single date.
   * * `range` - With calendar dropdown and a date range.
   */
  datePickerType: _propTypes.default.oneOf(['simple', 'single', 'range']),

  /**
   * Specify whether or not the input should be disabled
   */
  disabled: _propTypes.default.bool,

  /**
   * Provide text that is used alongside the control label for additional help
   */
  helperText: _propTypes.default.node,

  /**
   * Specify if the label should be hidden
   */
  hideLabel: _propTypes.default.bool,

  /**
   * The description of the calendar icon.
   */
  iconDescription: (0, _deprecate.default)(_propTypes.default.string, "\nThe prop `iconDescriptionInput` for DatePickerInput has been deprecated and is no longer used"),

  /**
   * Specify an id that uniquely identifies the `<input>`
   */
  id: _propTypes.default.string.isRequired,

  /**
   * Specify whether or not the input should be invalid
   */
  invalid: _propTypes.default.bool,

  /**
   * Specify the text to be rendered when the input is invalid
   */
  invalidText: _propTypes.default.node,

  /**
   * Provide the text that will be read by a screen reader when visiting this
   * control
   */
  labelText: _propTypes.default.node.isRequired,

  /**
   * Specify an `onChange` handler that is called whenever a change in the
   * input field has occurred
   */
  onChange: _propTypes.default.func,

  /**
   * Provide a function to be called when the input field is clicked
   */
  onClick: _propTypes.default.func,

  /**
   * Provide a function to be called when the input field is clicked
   */
  openCalendar: (0, _deprecate.default)(_propTypes.default.func, "\nThe prop `openCalendar` for DatePickerInput has been deprecated and is no longer used"),

  /**
   * Provide a regular expression that the input value must match
   */
  pattern: function pattern(props, propName, componentName) {
    if (props[propName] === undefined) {
      return;
    }

    try {
      new RegExp(props[propName]);
    } catch (e) {
      return new Error("Invalid value of prop '".concat(propName, "' supplied to '").concat(componentName, "', it should be a valid regular expression"));
    }
  },

  /**
   * Specify the placeholder text
   */
  placeholder: _propTypes.default.string,

  /**
   * Specify the size of the Date Picker Input. Currently supports either `sm`, `md`, or `lg` as an option.
   */
  size: _propTypes.default.oneOf(['sm', 'md', 'lg']),

  /**
   * Specify the type of the `<input>`
   */
  type: _propTypes.default.string,

  /**
   * Specify whether the control is currently in warning state
   */
  warn: _propTypes.default.bool,

  /**
   * Provide the text that is displayed when the control is in warning state
   */
  warnText: _propTypes.default.node
};

function DatePickerIcon(_ref) {
  var datePickerType = _ref.datePickerType,
      invalid = _ref.invalid,
      warn = _ref.warn,
      rest = (0, _objectWithoutProperties2.default)(_ref, _excluded3);
  var prefix = (0, _usePrefix.usePrefix)();

  if (datePickerType === 'simple' && !invalid && !warn) {
    return null;
  }

  if (invalid) {
    return /*#__PURE__*/_react.default.createElement(_iconsReact.WarningFilled16, {
      className: "".concat(prefix, "--date-picker__icon ").concat(prefix, "--date-picker__icon--invalid")
    });
  }

  if (!invalid && warn) {
    return /*#__PURE__*/_react.default.createElement(_iconsReact.WarningAltFilled16, {
      className: "".concat(prefix, "--date-picker__icon ").concat(prefix, "--date-picker__icon--warn")
    });
  }

  return /*#__PURE__*/_react.default.createElement(_iconsReact.Calendar16, {
    className: "".concat(prefix, "--date-picker__icon"),
    "aria-label": rest.iconDescription,
    onClick: rest.openCalendar,
    role: "img",
    "aria-hidden": "true"
  }, rest.iconDescription && /*#__PURE__*/_react.default.createElement("title", null, rest.iconDescription));
}

DatePickerIcon.propTypes = {
  /**
   * The type of the date picker:
   *
   * * `simple` - Without calendar dropdown.
   * * `single` - With calendar dropdown and single date.
   * * `range` - With calendar dropdown and a date range.
   */
  datePickerType: _propTypes.default.oneOf(['simple', 'single', 'range']),

  /**
   * Specify whether or not the input should be invalid
   */
  invalid: _propTypes.default.bool,

  /**
   * Specify whether the control is currently in warning state
   */
  warn: _propTypes.default.bool
};
var _default = DatePickerInput;
exports.default = _default;
