// carbon-components-react/es/components/Tooltip/next/DefinitionTooltip.stories.js
/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import './story.scss';
import React from 'react';
import { DefinitionTooltip } from './DefinitionTooltip';
import mdx from './DefinitionTooltip.mdx';
export default {
  title: 'Components/DefinitionTooltip',
  component: DefinitionTooltip,
  parameters: {
    controls: {
      hideNoControlsWarning: true
    },
    docs: {
      page: mdx
    },
    layout: 'centered'
  },
  argTypes: {
    children: {
      table: {
        disable: true
      }
    },
    className: {
      table: {
        disable: true
      }
    }
  },
  decorators: [function (Story) {
    return /*#__PURE__*/React.createElement("div", {
      className: "sb-tooltip-story sb-definition-tooltip"
    }, /*#__PURE__*/React.createElement(Story, null));
  }]
};
export var Default = function Default() {
  var definition = 'Uniform Resource Locator; the address of a resource (such as a document or website) on the Internet.';
  return /*#__PURE__*/React.createElement("p", null, "Custom domains direct requests for your apps in this Cloud Foundry organization to a", ' ', /*#__PURE__*/React.createElement(DefinitionTooltip, {
    definition: definition
  }, "URL"), " that you own. A custom domain can be a shared domain, a shared subdomain, or a shared domain and host.");
};

var PlaygroundStory = function PlaygroundStory(props) {
  var align = props.align,
      defaultOpen = props.defaultOpen,
      definition = props.definition;
  return /*#__PURE__*/React.createElement("p", null, "This sentence contains a", ' ', /*#__PURE__*/React.createElement(DefinitionTooltip, {
    align: align,
    defaultOpen: defaultOpen,
    definition: definition
  }, "definition tooltip"), ".");
};

export var Playground = PlaygroundStory.bind({});
Playground.argTypes = {
  align: {
    defaultValue: 'bottom-left',
    options: ['top', 'top-left', 'top-right', 'bottom', 'bottom-left', 'bottom-right'],
    control: {
      type: 'select'
    }
  },
  defaultOpen: {
    defaultValue: true
  },
  definition: {
    control: {
      type: 'text'
    },
    defaultValue: 'Example definition'
  }
};
