// carbon-components-react/lib/components/Accordion/Accordion-story.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.SkeletonPlayground = exports.Skeleton = exports.Playground = exports.AccordionStory = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _addonActions = require("@storybook/addon-actions");

var _addonKnobs = require("@storybook/addon-knobs");

var _Accordion = _interopRequireWildcard(require("../Accordion"));

var _Button = _interopRequireDefault(require("../Button"));

var _Accordion2 = _interopRequireDefault(require("./Accordion.mdx"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

/* eslint-disable no-console */
var _default = {
  title: 'Components/Accordion',
  component: _Accordion.default,
  subcomponents: {
    AccordionItem: _Accordion.AccordionItem,
    AccordionSkeleton: _Accordion.AccordionSkeleton
  },
  decorators: [_addonKnobs.withKnobs],
  parameters: {
    docs: {
      page: _Accordion2.default
    }
  }
};
exports.default = _default;

var AccordionStory = function AccordionStory() {
  return /*#__PURE__*/_react.default.createElement(_Accordion.default, null, /*#__PURE__*/_react.default.createElement(_Accordion.AccordionItem, {
    title: "Section 1 title"
  }, /*#__PURE__*/_react.default.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.")), /*#__PURE__*/_react.default.createElement(_Accordion.AccordionItem, {
    title: "Section 2 title"
  }, /*#__PURE__*/_react.default.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.")), /*#__PURE__*/_react.default.createElement(_Accordion.AccordionItem, {
    title: "Section 3 title"
  }, /*#__PURE__*/_react.default.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.")), /*#__PURE__*/_react.default.createElement(_Accordion.AccordionItem, {
    title: "Section 4 title"
  }, /*#__PURE__*/_react.default.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.")));
};

exports.AccordionStory = AccordionStory;
AccordionStory.storyName = 'Accordion';

var Skeleton = function Skeleton() {
  return /*#__PURE__*/_react.default.createElement(_Accordion.AccordionSkeleton, {
    open: true,
    count: 4
  });
};

exports.Skeleton = Skeleton;
Skeleton.decorators = [function (story) {
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      width: '500px'
    }
  }, story());
}];
var props = {
  onClick: (0, _addonActions.action)('onClick'),
  onHeadingClick: (0, _addonActions.action)('onHeadingClick')
};
var sizes = {
  'Small  (sm)': 'sm',
  'Medium (md) - default': undefined,
  'Large  (lg)': 'lg'
};

var Playground = function Playground() {
  return /*#__PURE__*/_react.default.createElement(_Accordion.default, {
    disabled: (0, _addonKnobs.boolean)('Disable entire Accordion (disabled)', false),
    size: (0, _addonKnobs.select)('Accordion heading size (size)', sizes, undefined) || undefined,
    align: (0, _addonKnobs.select)('Accordion heading alignment (align)', ['start', 'end'], 'end')
  }, /*#__PURE__*/_react.default.createElement(_Accordion.AccordionItem, (0, _extends2.default)({
    title: (0, _addonKnobs.text)('The title (title)', 'Section 1 title'),
    open: (0, _addonKnobs.boolean)('Open the section (open)', false)
  }, props), /*#__PURE__*/_react.default.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.")), /*#__PURE__*/_react.default.createElement(_Accordion.AccordionItem, (0, _extends2.default)({
    title: "Section 2 title"
  }, props), /*#__PURE__*/_react.default.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.")), /*#__PURE__*/_react.default.createElement(_Accordion.AccordionItem, (0, _extends2.default)({
    title: "Section 3 title"
  }, props, {
    disabled: (0, _addonKnobs.boolean)('Disable Section 3 (disabled)', true)
  }), /*#__PURE__*/_react.default.createElement(_Button.default, null, "This is a button.")), /*#__PURE__*/_react.default.createElement(_Accordion.AccordionItem, (0, _extends2.default)({
    title: /*#__PURE__*/_react.default.createElement("span", null, "Section 4 title (", /*#__PURE__*/_react.default.createElement("em", null, "the title can be a node"), ")")
  }, props), /*#__PURE__*/_react.default.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.")));
};

exports.Playground = Playground;

var SkeletonPlayground = function SkeletonPlayground() {
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      width: '500px'
    }
  }, /*#__PURE__*/_react.default.createElement(_Accordion.AccordionSkeleton, {
    align: (0, _addonKnobs.select)('Accordion heading alignment (align)', ['start', 'end'], 'end'),
    open: (0, _addonKnobs.boolean)('Show first item opened (open)', true),
    count: (0, _addonKnobs.number)('Set number of items (count)', 4)
  }));
};

exports.SkeletonPlayground = SkeletonPlayground;
