// carbon-components-react/es/components/Search/next/Search.stories.js
import _extends from "@babel/runtime/helpers/extends";

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import { ExpandableSearch } from 'carbon-components-react';
import Search from '../';
import React from 'react';
import { Layer } from '../../Layer';
export default {
  title: 'Components/Search',
  component: Search,
  argTypes: {
    size: {
      options: ['sm', 'md', 'lg'],
      control: {
        type: 'select'
      }
    }
  },
  subcomponents: {
    ExpandableSearch: ExpandableSearch
  }
};
export var Default = function Default(args) {
  return /*#__PURE__*/React.createElement(Search, _extends({
    size: "lg",
    defaultValue: "A default value",
    labelText: "Search",
    closeButtonLabelText: "Clear search input",
    id: "search-1",
    onChange: function onChange() {},
    onKeyDown: function onKeyDown() {}
  }, args));
};
export var Disabled = function Disabled(args) {
  return /*#__PURE__*/React.createElement(Search, _extends({
    disabled: true,
    size: "lg",
    defaultValue: "A default value",
    labelText: "Search",
    closeButtonLabelText: "Clear search input",
    id: "search-1",
    onChange: function onChange() {},
    onKeyDown: function onKeyDown() {}
  }, args));
};
export var Expandable = function Expandable(args) {
  return /*#__PURE__*/React.createElement(ExpandableSearch, _extends({
    size: "lg",
    labelText: "Search",
    closeButtonLabelText: "Clear search input",
    id: "search-expandable-1",
    onChange: function onChange() {},
    onKeyDown: function onKeyDown() {}
  }, args));
};
export var WithLayer = function WithLayer(args) {
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Search, _extends({
    size: "lg",
    defaultValue: "First Layer",
    labelText: "Search",
    closeButtonLabelText: "Clear search input",
    id: "search-1",
    onChange: function onChange() {},
    onKeyDown: function onKeyDown() {}
  }, args)), /*#__PURE__*/React.createElement(Layer, null, /*#__PURE__*/React.createElement(Search, _extends({
    size: "lg",
    defaultValue: "Second Layer",
    labelText: "Search",
    closeButtonLabelText: "Clear search input",
    id: "search-1",
    onChange: function onChange() {},
    onKeyDown: function onKeyDown() {}
  }, args)), /*#__PURE__*/React.createElement(Layer, null, /*#__PURE__*/React.createElement(Search, _extends({
    size: "lg",
    defaultValue: "Third Layer",
    labelText: "Search",
    closeButtonLabelText: "Clear search input",
    id: "search-1",
    onChange: function onChange() {},
    onKeyDown: function onKeyDown() {}
  }, args)))));
};
export var ExpandableWithLayer = function ExpandableWithLayer(args) {
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(ExpandableSearch, _extends({
    size: "lg",
    labelText: "First Layer",
    closeButtonLabelText: "Clear search input",
    id: "search-expandable-1",
    onChange: function onChange() {},
    onKeyDown: function onKeyDown() {}
  }, args)), /*#__PURE__*/React.createElement(Layer, null, /*#__PURE__*/React.createElement(ExpandableSearch, _extends({
    size: "lg",
    labelText: "Second Layer",
    closeButtonLabelText: "Clear search input",
    id: "search-expandable-1",
    onChange: function onChange() {},
    onKeyDown: function onKeyDown() {}
  }, args)), /*#__PURE__*/React.createElement(Layer, null, /*#__PURE__*/React.createElement(ExpandableSearch, _extends({
    size: "lg",
    labelText: "Third Layer",
    closeButtonLabelText: "Clear search input",
    id: "search-expandable-1",
    onChange: function onChange() {},
    onKeyDown: function onKeyDown() {}
  }, args)))));
};
