// carbon-components-react/es/components/DataTable/next/dynamic-content/DataTable-dynamic-content.stories.js
import _extends from "@babel/runtime/helpers/extends";
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _assertThisInitialized from "@babel/runtime/helpers/assertThisInitialized";
import _inherits from "@babel/runtime/helpers/inherits";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import _toConsumableArray from "@babel/runtime/helpers/toConsumableArray";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import './story.scss';
import React from 'react';
import { Delete16 as Delete, Save16 as Save, Download16 as Download } from '@carbon/icons-react';
import DataTable, { Table, TableBatchAction, TableBatchActions, TableBody, TableCell, TableContainer, TableExpandHeader, TableExpandRow, TableExpandedRow, TableHead, TableHeader, TableRow, TableSelectAll, TableSelectRow, TableToolbar, TableToolbarAction, TableToolbarContent, TableToolbarSearch, TableToolbarMenu } from '../..';
import { batchActionClick, rows, headers } from '../shared';
export default {
  title: 'Components/DataTable/Dynamic',
  component: DataTable,
  argTypes: {
    size: {
      options: ['xs', 'sm', 'md', 'lg', 'xl'],
      control: {
        type: 'select'
      }
    },
    useZebraStyles: {
      control: {
        type: 'boolean'
      }
    },
    radio: {
      control: {
        type: 'boolean'
      }
    },
    isSortable: {
      control: {
        type: 'boolean'
      }
    }
  },
  args: {
    size: 'lg',
    useZebraStyles: false
  }
};
export var Example = function Example(args) {
  var insertInRandomPosition = function insertInRandomPosition(array, element) {
    var index = Math.floor(Math.random() * (array.length + 1));
    return [].concat(_toConsumableArray(array.slice(0, index)), [element], _toConsumableArray(array.slice(index)));
  };

  var DynamicRows = /*#__PURE__*/function (_React$Component) {
    _inherits(DynamicRows, _React$Component);

    var _super = _createSuper(DynamicRows);

    function DynamicRows() {
      var _this;

      _classCallCheck(this, DynamicRows);

      for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }

      _this = _super.call.apply(_super, [this].concat(args));

      _defineProperty(_assertThisInitialized(_this), "state", {
        rows: rows,
        headers: headers,
        id: 0
      });

      _defineProperty(_assertThisInitialized(_this), "handleOnHeaderAdd", function () {
        var length = _this.state.headers.length;
        var header = {
          key: "header_".concat(length),
          header: "Header ".concat(length)
        };

        _this.setState(function (state) {
          var rows = state.rows.map(function (row) {
            return _objectSpread(_objectSpread({}, row), {}, _defineProperty({}, header.key, header.header));
          });
          return {
            rows: rows,
            headers: state.headers.concat(header)
          };
        });
      });

      _defineProperty(_assertThisInitialized(_this), "handleOnRowAdd", function () {
        _this.setState(function (state) {
          var _id = state.id,
              rows = state.rows;
          var id = _id + 1;
          var row = {
            id: '' + id,
            name: "New Row ".concat(id),
            protocol: 'HTTP',
            port: id * 100,
            rule: id % 2 === 0 ? 'Round robin' : 'DNS delegation',
            attached_groups: "Row ".concat(id, "'s VM Groups"),
            status: 'Starting'
          };
          state.headers.filter(function (header) {
            return row[header.key] === undefined;
          }).forEach(function (header) {
            row[header.key] = header.header;
          });
          return {
            id: id,
            rows: insertInRandomPosition(rows, row)
          };
        });
      });

      return _this;
    }

    _createClass(DynamicRows, [{
      key: "render",
      value: function render() {
        var _this2 = this;

        return /*#__PURE__*/React.createElement(DataTable, _extends({}, args, {
          rows: this.state.rows,
          headers: this.state.headers,
          render: function render(_ref) {
            var rows = _ref.rows,
                headers = _ref.headers,
                getHeaderProps = _ref.getHeaderProps,
                getSelectionProps = _ref.getSelectionProps,
                getToolbarProps = _ref.getToolbarProps,
                getBatchActionProps = _ref.getBatchActionProps,
                getRowProps = _ref.getRowProps,
                onInputChange = _ref.onInputChange,
                selectedRows = _ref.selectedRows,
                getTableProps = _ref.getTableProps,
                getTableContainerProps = _ref.getTableContainerProps;
            return /*#__PURE__*/React.createElement(TableContainer, _extends({
              title: "DataTable",
              description: "Use the toolbar menu to add rows and headers"
            }, getTableContainerProps()), /*#__PURE__*/React.createElement(TableToolbar, getToolbarProps(), /*#__PURE__*/React.createElement(TableBatchActions, getBatchActionProps(), /*#__PURE__*/React.createElement(TableBatchAction, {
              renderIcon: Delete,
              iconDescription: "Delete the selected rows",
              onClick: batchActionClick(selectedRows)
            }, "Delete"), /*#__PURE__*/React.createElement(TableBatchAction, {
              renderIcon: Save,
              iconDescription: "Save the selected rows",
              onClick: batchActionClick(selectedRows)
            }, "Save"), /*#__PURE__*/React.createElement(TableBatchAction, {
              renderIcon: Download,
              iconDescription: "Download the selected rows",
              onClick: batchActionClick(selectedRows)
            }, "Download")), /*#__PURE__*/React.createElement(TableToolbarContent, null, /*#__PURE__*/React.createElement(TableToolbarSearch, {
              onChange: onInputChange
            }), /*#__PURE__*/React.createElement(TableToolbarMenu, null, /*#__PURE__*/React.createElement(TableToolbarAction, {
              onClick: _this2.handleOnRowAdd
            }, "Add row"), /*#__PURE__*/React.createElement(TableToolbarAction, {
              onClick: _this2.handleOnHeaderAdd
            }, "Add header")))), /*#__PURE__*/React.createElement(Table, getTableProps(), /*#__PURE__*/React.createElement(TableHead, null, /*#__PURE__*/React.createElement(TableRow, null, /*#__PURE__*/React.createElement(TableExpandHeader, null), /*#__PURE__*/React.createElement(TableSelectAll, getSelectionProps()), headers.map(function (header, i) {
              return /*#__PURE__*/React.createElement(TableHeader, _extends({
                key: i
              }, getHeaderProps({
                header: header
              })), header.header);
            }))), /*#__PURE__*/React.createElement(TableBody, null, rows.map(function (row) {
              return /*#__PURE__*/React.createElement(React.Fragment, {
                key: row.id
              }, /*#__PURE__*/React.createElement(TableExpandRow, getRowProps({
                row: row
              }), /*#__PURE__*/React.createElement(TableSelectRow, getSelectionProps({
                row: row
              })), row.cells.map(function (cell) {
                return /*#__PURE__*/React.createElement(TableCell, {
                  key: cell.id
                }, cell.value);
              })), /*#__PURE__*/React.createElement(TableExpandedRow, {
                colSpan: headers.length + 3,
                className: "demo-expanded-td"
              }, /*#__PURE__*/React.createElement("h6", null, "Expandable row content"), /*#__PURE__*/React.createElement("div", null, "Description here")));
            }))));
          }
        }));
      }
    }]);

    return DynamicRows;
  }(React.Component);

  return /*#__PURE__*/React.createElement(DynamicRows, null);
};
