// carbon-components-react/lib/components/Tooltip/next/DefinitionTooltip.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Playground = exports.Default = void 0;

require("./story.scss");

var _react = _interopRequireDefault(require("react"));

var _DefinitionTooltip = require("./DefinitionTooltip");

var _DefinitionTooltip2 = _interopRequireDefault(require("./DefinitionTooltip.mdx"));

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
var _default = {
  title: 'Components/DefinitionTooltip',
  component: _DefinitionTooltip.DefinitionTooltip,
  parameters: {
    controls: {
      hideNoControlsWarning: true
    },
    docs: {
      page: _DefinitionTooltip2.default
    },
    layout: 'centered'
  },
  argTypes: {
    children: {
      table: {
        disable: true
      }
    },
    className: {
      table: {
        disable: true
      }
    }
  },
  decorators: [function (Story) {
    return /*#__PURE__*/_react.default.createElement("div", {
      className: "sb-tooltip-story sb-definition-tooltip"
    }, /*#__PURE__*/_react.default.createElement(Story, null));
  }]
};
exports.default = _default;

var Default = function Default() {
  var definition = 'Uniform Resource Locator; the address of a resource (such as a document or website) on the Internet.';
  return /*#__PURE__*/_react.default.createElement("p", null, "Custom domains direct requests for your apps in this Cloud Foundry organization to a", ' ', /*#__PURE__*/_react.default.createElement(_DefinitionTooltip.DefinitionTooltip, {
    definition: definition
  }, "URL"), " that you own. A custom domain can be a shared domain, a shared subdomain, or a shared domain and host.");
};

exports.Default = Default;

var PlaygroundStory = function PlaygroundStory(props) {
  var align = props.align,
      defaultOpen = props.defaultOpen,
      definition = props.definition;
  return /*#__PURE__*/_react.default.createElement("p", null, "This sentence contains a", ' ', /*#__PURE__*/_react.default.createElement(_DefinitionTooltip.DefinitionTooltip, {
    align: align,
    defaultOpen: defaultOpen,
    definition: definition
  }, "definition tooltip"), ".");
};

var Playground = PlaygroundStory.bind({});
exports.Playground = Playground;
Playground.argTypes = {
  align: {
    defaultValue: 'bottom-left',
    options: ['top', 'top-left', 'top-right', 'bottom', 'bottom-left', 'bottom-right'],
    control: {
      type: 'select'
    }
  },
  defaultOpen: {
    defaultValue: true
  },
  definition: {
    control: {
      type: 'text'
    },
    defaultValue: 'Example definition'
  }
};
