// carbon-components-react/lib/components/TextArea/next/TextArea.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithLayer = exports.Skeleton = exports.Default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = _interopRequireWildcard(require("../"));

var _Layer = require("../../Layer");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
var _default = {
  title: 'Components/TextArea',
  component: _.default,
  subcomponents: {
    TextAreaSkeleton: _.TextAreaSkeleton
  }
};
exports.default = _default;

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_.default, {
    labelText: "Text Area label",
    helperText: "Optional helper text",
    cols: 50,
    rows: 4,
    id: "text-area-1"
  });
};

exports.Default = Default;

var WithLayer = function WithLayer() {
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_.default, {
    labelText: "First layer",
    helperText: "Optional helper text",
    cols: 50,
    rows: 4,
    id: "text-area-1"
  }), /*#__PURE__*/_react.default.createElement(_Layer.Layer, null, /*#__PURE__*/_react.default.createElement(_.default, {
    labelText: "Second layer",
    helperText: "Optional helper text",
    cols: 50,
    rows: 4,
    id: "text-area-1"
  }), /*#__PURE__*/_react.default.createElement(_Layer.Layer, null, /*#__PURE__*/_react.default.createElement(_.default, {
    labelText: "Third layer",
    helperText: "Optional helper text",
    cols: 50,
    rows: 4,
    id: "text-area-1"
  }))));
};

exports.WithLayer = WithLayer;

var Skeleton = function Skeleton() {
  return /*#__PURE__*/_react.default.createElement(_.TextAreaSkeleton, null);
};

exports.Skeleton = Skeleton;
