// carbon-components-react/lib/components/Tile/index.js
"use strict";

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TileBelowTheFoldContent = exports.TileAboveTheFoldContent = exports.Tile = exports.SelectableTile = exports.ExpandableTile = exports.ClickableTile = void 0;

var FeatureFlags = _interopRequireWildcard(require("@carbon/feature-flags"));

var _Tile = require("./next/Tile");

var _Tile2 = require("./Tile");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Tile = FeatureFlags.enabled('enable-v11-release') ? _Tile.Tile : _Tile2.Tile;
exports.Tile = Tile;
var ClickableTile = FeatureFlags.enabled('enable-v11-release') ? _Tile.ClickableTile : _Tile2.ClickableTile;
exports.ClickableTile = ClickableTile;
var SelectableTile = FeatureFlags.enabled('enable-v11-release') ? _Tile.SelectableTile : _Tile2.SelectableTile;
exports.SelectableTile = SelectableTile;
var ExpandableTile = FeatureFlags.enabled('enable-v11-release') ? _Tile.ExpandableTile : _Tile2.ExpandableTile;
exports.ExpandableTile = ExpandableTile;
var TileAboveTheFoldContent = FeatureFlags.enabled('enable-v11-release') ? _Tile.TileAboveTheFoldContent : _Tile2.TileAboveTheFoldContent;
exports.TileAboveTheFoldContent = TileAboveTheFoldContent;
var TileBelowTheFoldContent = FeatureFlags.enabled('enable-v11-release') ? _Tile.TileBelowTheFoldContent : _Tile2.TileBelowTheFoldContent;
exports.TileBelowTheFoldContent = TileBelowTheFoldContent;
