// carbon-components-react/lib/components/Tag/next/Tag.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Default = void 0;

var _react = _interopRequireDefault(require("react"));

var _carbonComponentsReact = require("carbon-components-react");

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
var _default = {
  title: 'Components/Tag',
  component: _carbonComponentsReact.Tag
};
exports.default = _default;

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_carbonComponentsReact.Tag, {
    className: "some-class",
    type: "red",
    size: "sm",
    title: "Clear Filter"
  }, 'Tag content'), /*#__PURE__*/_react.default.createElement(_carbonComponentsReact.Tag, {
    className: "some-class",
    type: "magenta",
    size: "sm",
    title: "Clear Filter"
  }, 'Tag content'), /*#__PURE__*/_react.default.createElement(_carbonComponentsReact.Tag, {
    className: "some-class",
    type: "purple",
    size: "sm",
    title: "Clear Filter"
  }, 'Tag content'), /*#__PURE__*/_react.default.createElement(_carbonComponentsReact.Tag, {
    className: "some-class",
    type: "blue",
    size: "sm",
    title: "Clear Filter"
  }, 'Tag content'), /*#__PURE__*/_react.default.createElement(_carbonComponentsReact.Tag, {
    className: "some-class",
    type: "cyan",
    size: "sm",
    title: "Clear Filter"
  }, 'Tag content'), /*#__PURE__*/_react.default.createElement(_carbonComponentsReact.Tag, {
    className: "some-class",
    type: "teal",
    size: "sm",
    title: "Clear Filter"
  }, 'Tag content'), /*#__PURE__*/_react.default.createElement(_carbonComponentsReact.Tag, {
    className: "some-class",
    type: "green",
    size: "sm",
    title: "Clear Filter"
  }, 'Tag content'), /*#__PURE__*/_react.default.createElement(_carbonComponentsReact.Tag, {
    className: "some-class",
    type: "gray",
    size: "sm",
    title: "Clear Filter"
  }, 'Tag content'), /*#__PURE__*/_react.default.createElement(_carbonComponentsReact.Tag, {
    className: "some-class",
    type: "cool-gray",
    size: "sm",
    title: "Clear Filter"
  }, 'Tag content'), /*#__PURE__*/_react.default.createElement(_carbonComponentsReact.Tag, {
    className: "some-class",
    type: "warm-gray",
    size: "sm",
    title: "Clear Filter"
  }, 'Tag content'), /*#__PURE__*/_react.default.createElement(_carbonComponentsReact.Tag, {
    className: "some-class",
    type: "high-contrast",
    size: "sm",
    title: "Clear Filter"
  }, 'Tag content'), /*#__PURE__*/_react.default.createElement(_carbonComponentsReact.Tag, {
    className: "some-class",
    type: "outline",
    size: "sm",
    title: "Clear Filter"
  }, 'Tag content'));
};

exports.Default = Default;
