// carbon-components-react/lib/components/Layer/next/Layer.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Default = void 0;

require("./Layer-story.scss");

var _react = _interopRequireDefault(require("react"));

var _Layer = require("../../Layer");

var _Layer2 = _interopRequireDefault(require("./Layer.mdx"));

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
var _default = {
  title: 'Components/Layer',
  component: _Layer.Layer,
  parameters: {
    controls: {
      hideNoControlsWarning: true
    },
    docs: {
      page: _Layer2.default
    }
  },
  argTypes: {
    as: {
      table: {
        disable: true
      }
    },
    children: {
      table: {
        disable: true
      }
    },
    className: {
      table: {
        disable: true
      }
    }
  }
};
exports.default = _default;

var Default = function Default() {
  function TestComponent() {
    return /*#__PURE__*/_react.default.createElement("div", {
      className: "example-layer-test-component"
    }, "Test component");
  }

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(TestComponent, null), /*#__PURE__*/_react.default.createElement(_Layer.Layer, null, /*#__PURE__*/_react.default.createElement(TestComponent, null), /*#__PURE__*/_react.default.createElement(_Layer.Layer, null, /*#__PURE__*/_react.default.createElement(TestComponent, null))));
};

exports.Default = Default;
