// carbon-components-react/lib/components/DatePicker/next/DatePicker.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof3 = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _typeof2 = _interopRequireDefault(require("@babel/runtime/helpers/typeof"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _flatpickr2 = _interopRequireDefault(require("flatpickr"));

var _index = _interopRequireDefault(require("flatpickr/dist/l10n/index"));

var _DatePickerInput = _interopRequireDefault(require("../../DatePickerInput"));

var _appendToPlugin = _interopRequireDefault(require("../plugins/appendToPlugin"));

var _fixEventsPlugin = _interopRequireDefault(require("../plugins/fixEventsPlugin"));

var _rangePlugin = _interopRequireDefault(require("../plugins/rangePlugin"));

var _keyboard = require("../../../internal/keyboard");

var _usePrefix = require("../../../internal/usePrefix");

var _useSavedCallback = require("../../../internal/useSavedCallback");

var _excluded = ["allowInput", "appendTo", "children", "className", "dateFormat", "datePickerType", "disable", "enable", "inline", "light", "locale", "maxDate", "minDate", "onChange", "onClose", "onOpen", "short", "value"];

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof3(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

// Weekdays shorthand for english locale
_index.default.en.weekdays.shorthand.forEach(function (_day, index) {
  var currentDay = _index.default.en.weekdays.shorthand;

  if (currentDay[index] === 'Thu' || currentDay[index] === 'Th') {
    currentDay[index] = 'Th';
  } else {
    currentDay[index] = currentDay[index].charAt(0);
  }
});

var forEach = Array.prototype.forEach;
/**
 * @param {number} monthNumber The month number.
 * @param {boolean} shorthand `true` to use shorthand month.
 * @param {Locale} locale The Flatpickr locale data.
 * @returns {string} The month string.
 */

var monthToStr = function monthToStr(monthNumber, shorthand, locale) {
  return locale.months[shorthand ? 'shorthand' : 'longhand'][monthNumber];
};
/**
 * @param {object} config Plugin configuration.
 * @param {boolean} [config.shorthand] `true` to use shorthand month.
 * @param {string} config.selectorFlatpickrMonthYearContainer The CSS selector for the container of month/year selection UI.
 * @param {string} config.selectorFlatpickrYearContainer The CSS selector for the container of year selection UI.
 * @param {string} config.selectorFlatpickrCurrentMonth The CSS selector for the text-based month selection UI.
 * @param {string} config.classFlatpickrCurrentMonth The CSS class for the text-based month selection UI.
 * @returns {Plugin} A Flatpickr plugin to use text instead of `<select>` for month picker.
 */


var carbonFlatpickrMonthSelectPlugin = function carbonFlatpickrMonthSelectPlugin(config) {
  return function (fp) {
    var setupElements = function setupElements() {
      var _fp$monthElements;

      if (!fp.monthElements) {
        return;
      }

      fp.monthElements.forEach(function (elem) {
        if (!elem.parentNode) {
          return;
        }

        elem.parentNode.removeChild(elem);
      });

      (_fp$monthElements = fp.monthElements).splice.apply(_fp$monthElements, [0, fp.monthElements.length].concat((0, _toConsumableArray2.default)(fp.monthElements.map(function () {
        // eslint-disable-next-line no-underscore-dangle
        var monthElement = fp._createElement('span', config.classFlatpickrCurrentMonth);

        monthElement.textContent = monthToStr(fp.currentMonth, config.shorthand === true, fp.l10n);
        fp.yearElements[0].closest(config.selectorFlatpickrMonthYearContainer).insertBefore(monthElement, fp.yearElements[0].closest(config.selectorFlatpickrYearContainer));
        return monthElement;
      }))));
    };

    var updateCurrentMonth = function updateCurrentMonth() {
      var monthStr = monthToStr(fp.currentMonth, config.shorthand === true, fp.l10n);
      fp.yearElements.forEach(function (elem) {
        var currentMonthContainer = elem.closest(config.selectorFlatpickrMonthYearContainer);
        Array.prototype.forEach.call(currentMonthContainer.querySelectorAll('.cur-month'), function (monthElement) {
          monthElement.textContent = monthStr;
        });
      });
    };

    var register = function register() {
      fp.loadedPlugins.push('carbonFlatpickrMonthSelectPlugin');
    };

    return {
      onMonthChange: updateCurrentMonth,
      onValueUpdate: updateCurrentMonth,
      onOpen: updateCurrentMonth,
      onReady: [setupElements, updateCurrentMonth, register]
    };
  };
};
/**
 * Determine if every child in a list of children has no label specified
 * @param {Array<ReactElement>} children
 * @returns {boolean}
 */


function isLabelTextEmpty(children) {
  return children.every(function (child) {
    return !child.props.labelText;
  });
}

var rightArrowHTML = "<svg width=\"16px\" height=\"16px\" viewBox=\"0 0 16 16\">\n  <polygon points=\"11,8 6,13 5.3,12.3 9.6,8 5.3,3.7 6,3 \"/>\n  <rect width=\"16\" height=\"16\" style=\"fill:none\" />\n</svg>";
var leftArrowHTML = "<svg width=\"16px\" height=\"16px\" viewBox=\"0 0 16 16\">\n  <polygon points=\"5,8 10,3 10.7,3.7 6.4,8 10.7,12.3 10,13 \"/>\n  <rect width=\"16\" height=\"16\" style=\"fill:none\" />\n</svg>";

function updateClassNames(calendar, prefix) {
  var calendarContainer = calendar.calendarContainer;
  var daysContainer = calendar.days;

  if (calendarContainer && daysContainer) {
    // calendarContainer and daysContainer are undefined if flatpickr detects a mobile device
    calendarContainer.classList.add("".concat(prefix, "--date-picker__calendar"));
    calendarContainer.querySelector('.flatpickr-month').classList.add("".concat(prefix, "--date-picker__month"));
    calendarContainer.querySelector('.flatpickr-weekdays').classList.add("".concat(prefix, "--date-picker__weekdays"));
    calendarContainer.querySelector('.flatpickr-days').classList.add("".concat(prefix, "--date-picker__days"));
    forEach.call(calendarContainer.querySelectorAll('.flatpickr-weekday'), function (item) {
      var currentItem = item;
      currentItem.innerHTML = currentItem.innerHTML.replace(/\s+/g, '');
      currentItem.classList.add("".concat(prefix, "--date-picker__weekday"));
    });
    forEach.call(daysContainer.querySelectorAll('.flatpickr-day'), function (item) {
      item.classList.add("".concat(prefix, "--date-picker__day"));

      if (item.classList.contains('today') && calendar.selectedDates.length > 0) {
        item.classList.add('no-border');
      } else if (item.classList.contains('today') && calendar.selectedDates.length === 0) {
        item.classList.remove('no-border');
      }
    });
  }
}

function DatePicker(_ref) {
  var _cx;

  var allowInput = _ref.allowInput,
      appendTo = _ref.appendTo,
      children = _ref.children,
      className = _ref.className,
      _ref$dateFormat = _ref.dateFormat,
      dateFormat = _ref$dateFormat === void 0 ? 'm/d/Y' : _ref$dateFormat,
      datePickerType = _ref.datePickerType,
      disable = _ref.disable,
      enable = _ref.enable,
      inline = _ref.inline,
      _ref$light = _ref.light,
      light = _ref$light === void 0 ? false : _ref$light,
      _ref$locale = _ref.locale,
      locale = _ref$locale === void 0 ? 'en' : _ref$locale,
      maxDate = _ref.maxDate,
      minDate = _ref.minDate,
      onChange = _ref.onChange,
      onClose = _ref.onClose,
      onOpen = _ref.onOpen,
      _ref$short = _ref.short,
      short = _ref$short === void 0 ? false : _ref$short,
      value = _ref.value,
      rest = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  var prefix = (0, _usePrefix.usePrefix)();
  var startInputField = (0, _react.useRef)(null);
  var endInputField = (0, _react.useRef)(null);
  var calendarRef = (0, _react.useRef)(null);
  var savedOnChange = (0, _useSavedCallback.useSavedCallback)(onChange);
  var savedOnClose = (0, _useSavedCallback.useSavedCallback)(onClose);
  var savedOnOpen = (0, _useSavedCallback.useSavedCallback)(onOpen);
  var datePickerClasses = (0, _classnames.default)("".concat(prefix, "--date-picker"), (_cx = {}, (0, _defineProperty2.default)(_cx, "".concat(prefix, "--date-picker--short"), short), (0, _defineProperty2.default)(_cx, "".concat(prefix, "--date-picker--light"), light), (0, _defineProperty2.default)(_cx, "".concat(prefix, "--date-picker--simple"), datePickerType === 'simple'), (0, _defineProperty2.default)(_cx, "".concat(prefix, "--date-picker--single"), datePickerType === 'single'), (0, _defineProperty2.default)(_cx, "".concat(prefix, "--date-picker--range"), datePickerType === 'range'), (0, _defineProperty2.default)(_cx, "".concat(prefix, "--date-picker--nolabel"), datePickerType === 'range' && isLabelTextEmpty(children)), _cx));
  var wrapperClasses = (0, _classnames.default)("".concat(prefix, "--form-item"), (0, _defineProperty2.default)({}, className, className));

  var childrenWithProps = _react.default.Children.toArray(children).map(function (child, index) {
    if (index === 0 && child.type === /*#__PURE__*/_react.default.createElement(_DatePickerInput.default, child.props).type) {
      return /*#__PURE__*/_react.default.cloneElement(child, {
        datePickerType: datePickerType,
        ref: startInputField,
        openCalendar: function openCalendar() {
          if (calendarRef.current) {
            calendarRef.current.open();
          }
        }
      });
    }

    if (index === 1 && child.type === /*#__PURE__*/_react.default.createElement(_DatePickerInput.default, child.props).type) {
      return /*#__PURE__*/_react.default.cloneElement(child, {
        datePickerType: datePickerType,
        ref: endInputField,
        openCalendar: function openCalendar() {
          if (calendarRef.current) {
            calendarRef.current.open();
          }
        }
      });
    }

    if (index === 0) {
      return /*#__PURE__*/_react.default.cloneElement(child, {
        ref: startInputField
      });
    }

    if (index === 1) {
      return /*#__PURE__*/_react.default.cloneElement(child, {
        ref: endInputField
      });
    }
  });

  (0, _react.useEffect)(function () {
    var _flatpickr;

    if (datePickerType !== 'single' && datePickerType !== 'range') {
      return;
    }

    if (startInputField.current === null) {
      return;
    }

    var onHook = function onHook(_electedDates, _dateStr, instance, prefix) {
      updateClassNames(instance, prefix);
    }; // Logic to determine if `enable` or `disable` will be passed down. If neither
    // is provided, we return the default empty disabled array, allowing all dates.


    var enableOrDisable = enable ? 'enable' : 'disable';
    var enableOrDisableArr;

    if (!enable && !disable) {
      enableOrDisableArr = [];
    } else if (enable) {
      enableOrDisableArr = enable;
    } else {
      enableOrDisableArr = disable;
    }

    var localeData;

    if ((0, _typeof2.default)(locale) === 'object') {
      var location = locale.locale ? locale.locale : 'en';
      localeData = _objectSpread(_objectSpread({}, _index.default[location]), locale);
    } else {
      localeData = _index.default[locale];
    }

    var start = startInputField.current;
    var end = endInputField.current;
    var calendar = new _flatpickr2.default(start, (_flatpickr = {
      inline: inline !== null && inline !== void 0 ? inline : false,
      disableMobile: true,
      defaultDate: value,
      mode: datePickerType,
      allowInput: allowInput !== null && allowInput !== void 0 ? allowInput : true,
      dateFormat: dateFormat,
      locale: localeData
    }, (0, _defineProperty2.default)(_flatpickr, enableOrDisable, enableOrDisableArr), (0, _defineProperty2.default)(_flatpickr, "minDate", minDate), (0, _defineProperty2.default)(_flatpickr, "maxDate", maxDate), (0, _defineProperty2.default)(_flatpickr, "plugins", [datePickerType === 'range' ? new _rangePlugin.default({
      input: endInputField.current
    }) : function () {}, appendTo ? (0, _appendToPlugin.default)({
      appendTo: appendTo
    }) : function () {}, carbonFlatpickrMonthSelectPlugin({
      selectorFlatpickrMonthYearContainer: '.flatpickr-current-month',
      selectorFlatpickrYearContainer: '.numInputWrapper',
      selectorFlatpickrCurrentMonth: '.cur-month',
      classFlatpickrCurrentMonth: 'cur-month'
    }), (0, _fixEventsPlugin.default)({
      inputFrom: startInputField.current,
      inputTo: endInputField.current
    })]), (0, _defineProperty2.default)(_flatpickr, "clickOpens", true), (0, _defineProperty2.default)(_flatpickr, "nextArrow", rightArrowHTML), (0, _defineProperty2.default)(_flatpickr, "prevArrow", leftArrowHTML), (0, _defineProperty2.default)(_flatpickr, "onChange", savedOnChange), (0, _defineProperty2.default)(_flatpickr, "onClose", savedOnClose), (0, _defineProperty2.default)(_flatpickr, "onReady", onHook), (0, _defineProperty2.default)(_flatpickr, "onMonthChange", onHook), (0, _defineProperty2.default)(_flatpickr, "onYearChange", onHook), (0, _defineProperty2.default)(_flatpickr, "onOpen", function onOpen() {
      onHook.apply(void 0, arguments);
      savedOnOpen.apply(void 0, arguments);
    }), (0, _defineProperty2.default)(_flatpickr, "onValueUpdate", onHook), _flatpickr));
    calendarRef.current = calendar;

    function handleArrowDown(event) {
      if ((0, _keyboard.match)(event, _keyboard.keys.ArrowDown)) {
        var calendarContainer = calendar.calendarContainer,
            fpSelectedDateElem = calendar.selectedDateElem,
            fptodayDateElem = calendar.todayDateElem;
        var selectedDateElem = calendarContainer.querySelector('.selected') && fpSelectedDateElem;
        var todayDateElem = calendarContainer.querySelector('.today') && fptodayDateElem;
        (selectedDateElem || todayDateElem || calendarContainer.querySelector('.flatpickr-day[tabindex]') || calendarContainer).focus();
      }
    }

    function handleOnChange() {
      if (start.value !== '') {
        return;
      }

      if (!calendar.selectedDates) {
        return;
      }

      if (calendar.selectedDates.length === 0) {
        return;
      }

      calendar.clear();
      calendar.input.focus();
    }

    if (start) {
      start.addEventListener('keydown', handleArrowDown);
      start.addEventListener('change', handleOnChange); // Flatpickr's calendar dialog is not rendered in a landmark causing an
      // error with IBM Equal Access Accessibility Checker so we add an aria
      // role to the container div.

      calendar.calendarContainer.setAttribute('role', 'region'); // IBM EAAC requires an aria-label on a role='region'

      calendar.calendarContainer.setAttribute('aria-label', 'calendar-container');
    }

    if (end) {
      end.addEventListener('keydown', handleArrowDown);
      end.addEventListener('change', handleOnChange);
    } //component did unmount equivalent


    return function () {
      // Note: if the `startInputField` ref is undefined then calendar will be
      // of type: Array and `destroy` will not be defined
      if (calendar && calendar.destroy) {
        calendar.destroy();
      }

      if (start) {
        start.removeEventListener('keydown', handleArrowDown);
        start.removeEventListener('change', handleOnChange);
      }

      if (end) {
        end.removeEventListener('keydown', handleArrowDown);
        end.removeEventListener('change', handleOnChange);
      }
    };
  }, [savedOnChange, savedOnClose, savedOnOpen]); //eslint-disable-line react-hooks/exhaustive-deps

  (0, _react.useEffect)(function () {
    if (calendarRef.current) {
      calendarRef.current.set({
        dateFormat: dateFormat
      });
    }
  }, [dateFormat]);
  (0, _react.useEffect)(function () {
    if (calendarRef.current && minDate) {
      calendarRef.current.set('minDate', minDate);
    }
  }, [minDate]);
  (0, _react.useEffect)(function () {
    if (calendarRef.current && maxDate) {
      calendarRef.current.set('maxDate', maxDate);
    }
  }, [maxDate]);
  (0, _react.useEffect)(function () {
    if (calendarRef.current && disable) {
      calendarRef.current.set('disbale', disable);
    }
  }, [disable]);
  (0, _react.useEffect)(function () {
    if (calendarRef.current && enable) {
      calendarRef.current.set('enable', enable);
    }
  }, [enable]);
  (0, _react.useEffect)(function () {
    if (calendarRef.current && inline) {
      calendarRef.current.set('inline', inline);
    }
  }, [inline]);
  (0, _react.useEffect)(function () {
    if (calendarRef.current) {
      calendarRef.current.set({
        value: value
      });
      updateClassNames(calendarRef.current, prefix); //for simple date picker w/o calendar; initial mount may not have value
    } else if (!calendarRef.current && value) {
      startInputField.current.value = value;
    }
  }, [value, prefix]);
  return /*#__PURE__*/_react.default.createElement("div", (0, _extends2.default)({
    className: wrapperClasses
  }, rest), /*#__PURE__*/_react.default.createElement("div", {
    className: datePickerClasses
  }, childrenWithProps));
}

DatePicker.propTypes = {
  /**
   * flatpickr prop passthrough. Allows the user to enter a date directly
   * into the input field
   */
  allowInput: _propTypes.default.bool,

  /**
   * The DOM element the Flatpicker should be inserted into. `<body>` by default.
   */
  appendTo: _propTypes.default.object,

  /**
   * The child nodes.
   */
  children: _propTypes.default.node,

  /**
   * The CSS class names.
   */
  className: _propTypes.default.string,

  /**
   * The date format.
   */
  dateFormat: _propTypes.default.string,

  /**
   * The type of the date picker:
   *
   * * `simple` - Without calendar dropdown.
   * * `single` - With calendar dropdown and single date.
   * * `range` - With calendar dropdown and a date range.
   */
  datePickerType: _propTypes.default.oneOf(['simple', 'single', 'range']),

  /**
   * The flatpickr `disable` option that allows a user to disable certain dates.
   */
  disable: _propTypes.default.array,

  /**
   * The flatpickr `enable` option that allows a user to enable certain dates.
   */
  enable: _propTypes.default.array,

  /**
   * The flatpickr `inline` option.
   */
  inline: _propTypes.default.bool,

  /**
   * `true` to use the light version.
   */
  light: _propTypes.default.bool,

  /**
   *  The language locale used to format the days of the week, months, and numbers. The full list of supported locales can be found here https://github.com/flatpickr/flatpickr/tree/master/src/l10n
   */
  locale: _propTypes.default.oneOfType([_propTypes.default.object, _propTypes.default.oneOf(['ar', // Arabic
  'at', // Austria
  'az', // Azerbaijan
  'be', // Belarusian
  'bg', // Bulgarian
  'bn', // Bangla
  'bs', // Bosnia
  'cat', // Catalan
  'cs', // Czech
  'cy', // Welsh
  'da', // Danish
  'de', // German
  'en', // English
  'eo', // Esperanto
  'es', // Spanish
  'et', // Estonian
  'fa', // Persian
  'fi', // Finnish
  'fo', // Faroese
  'fr', // French
  'ga', // Gaelic
  'gr', // Greek
  'he', // Hebrew
  'hi', // Hindi
  'hr', // Croatian
  'hu', // Hungarian
  'id', // Indonesian
  'is', // Icelandic
  'it', // Italian
  'ja', // Japanese
  'ka', // Georgian
  'km', // Khmer
  'ko', // Korean
  'kz', // Kazakh
  'lt', // Lithuanian
  'lv', // Latvian
  'mk', // Macedonian
  'mn', // Mongolian
  'ms', // Malaysian
  'my', // Burmese
  'nl', // Dutch
  'no', // Norwegian
  'pa', // Punjabi
  'pl', // Polish
  'pt', // Portuguese
  'ro', // Romanian
  'ru', // Russian
  'si', // Sinhala
  'sk', // Slovak
  'sl', // Slovenian
  'sq', // Albanian
  'sr', // Serbian
  'sv', // Swedish
  'th', // Thai
  'tr', // Turkish
  'uk', // Ukrainian
  'uz', // Uzbek
  'uz_latn', // Uzbek Latin
  'vn', // Vietnamese
  'zh_tw', // Mandarin Traditional
  'zh' // Mandarin
  ])]),

  /**
   * The maximum date that a user can pick to.
   */
  maxDate: _propTypes.default.string,

  /**
   * The minimum date that a user can start picking from.
   */
  minDate: _propTypes.default.string,

  /**
   * The `change` event handler.
   */
  onChange: _propTypes.default.func,

  /**
   * The `close` event handler.
   */
  onClose: _propTypes.default.func,

  /**
   * The `open` event handler.
   */
  onOpen: _propTypes.default.func,

  /**
   * `true` to use the short version.
   */
  short: _propTypes.default.bool,

  /**
   * The value of the date value provided to flatpickr, could
   * be a date, a date number, a date string, an array of dates.
   */
  value: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.arrayOf(_propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.number, _propTypes.default.object])), _propTypes.default.object, _propTypes.default.number])
};
var _default = DatePicker;
exports.default = _default;
