// carbon-components-react/lib/components/Pagination/Pagination-story.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports._Pagination = exports.PaginationWithCustomPageSizesLabel = exports.MultiplePaginationComponents = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _addonActions = require("@storybook/addon-actions");

var _addonKnobs = require("@storybook/addon-knobs");

var _Pagination2 = _interopRequireDefault(require("../Pagination"));

var _Pagination3 = _interopRequireDefault(require("./Pagination.mdx"));

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
var sizes = {
  'Small  (sm)': 'sm',
  'Medium (md) - default': undefined,
  'Large  (lg)': 'lg'
};

var props = function props() {
  return {
    disabled: (0, _addonKnobs.boolean)('Disable page inputs (disabled)', false),
    size: (0, _addonKnobs.select)('Size (size)', sizes, undefined) || undefined,
    page: (0, _addonKnobs.number)('The current page (page)', 1),
    totalItems: (0, _addonKnobs.number)('Total number of items (totalItems)', 103),
    pagesUnknown: (0, _addonKnobs.boolean)('Total number of items unknown (pagesUnknown)', false),
    pageInputDisabled: (0, _addonKnobs.boolean)('Disable page input (pageInputDisabled)', undefined),
    pageSizeInputDisabled: (0, _addonKnobs.boolean)('Disable page size input (pageSizeInputDisabled)', undefined),
    backwardText: (0, _addonKnobs.text)('The description for the backward icon (backwardText)', 'Previous page'),
    forwardText: (0, _addonKnobs.text)('The description for the forward icon (forwardText)', 'Next page'),
    pageSize: (0, _addonKnobs.number)('Number of items per page (pageSize)', 10),
    pageSizes: (0, _addonKnobs.array)('Choices of `pageSize` (pageSizes)', [10, 20, 30, 40, 50]),
    itemsPerPageText: (0, _addonKnobs.text)('Label for `pageSizes` select UI (itemsPerPageText)', 'Items per page:'),
    onChange: (0, _addonActions.action)('onChange')
  };
};

var _default = {
  title: 'Components/Pagination',
  component: _Pagination2.default,
  decorators: [_addonKnobs.withKnobs, function (story) {
    return /*#__PURE__*/_react.default.createElement("div", {
      style: {
        maxWidth: '800px'
      }
    }, story());
  }],
  parameters: {
    docs: {
      page: _Pagination3.default
    }
  }
};
exports.default = _default;

var _Pagination = function _Pagination() {
  return /*#__PURE__*/_react.default.createElement(_Pagination2.default, props());
};

exports._Pagination = _Pagination;
_Pagination.parameters = {
  info: {
    text: "\n            The pagination component is used to switch through multiple pages of items, when only a maximum number of items can be displayed per page. Can be used in combination with other components like DataTable.\n          "
  }
};

var MultiplePaginationComponents = function MultiplePaginationComponents() {
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_Pagination2.default, props()), /*#__PURE__*/_react.default.createElement(_Pagination2.default, props()));
};

exports.MultiplePaginationComponents = MultiplePaginationComponents;
MultiplePaginationComponents.storyName = 'Multiple Pagination components';
MultiplePaginationComponents.parameters = {
  info: {
    text: "Showcasing unique ids for each pagination component"
  }
};

var PaginationWithCustomPageSizesLabel = function PaginationWithCustomPageSizesLabel() {
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_Pagination2.default, (0, _extends2.default)({}, props(), {
    pageSizes: [{
      text: 'Ten',
      value: 10
    }, {
      text: 'Twenty',
      value: 20
    }, {
      text: 'Thirty',
      value: 30
    }, {
      text: 'Forty',
      value: 40
    }, {
      text: 'Fifty',
      value: 50
    }]
  })));
};

exports.PaginationWithCustomPageSizesLabel = PaginationWithCustomPageSizesLabel;
PaginationWithCustomPageSizesLabel.storyName = 'Pagination with custom page sizes label';
PaginationWithCustomPageSizesLabel.parameters = {
  info: {
    text: "It is also possible to use custom label in page sizes list"
  }
};
