// carbon-components-react/lib/components/Search/next/Search.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithLayer = exports.ExpandableWithLayer = exports.Expandable = exports.Disabled = exports.Default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _carbonComponentsReact = require("carbon-components-react");

var _ = _interopRequireDefault(require("../"));

var _react = _interopRequireDefault(require("react"));

var _Layer = require("../../Layer");

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
var _default = {
  title: 'Components/Search',
  component: _.default,
  argTypes: {
    size: {
      options: ['sm', 'md', 'lg'],
      control: {
        type: 'select'
      }
    }
  },
  subcomponents: {
    ExpandableSearch: _carbonComponentsReact.ExpandableSearch
  }
};
exports.default = _default;

var Default = function Default(args) {
  return /*#__PURE__*/_react.default.createElement(_.default, (0, _extends2.default)({
    size: "lg",
    defaultValue: "A default value",
    labelText: "Search",
    closeButtonLabelText: "Clear search input",
    id: "search-1",
    onChange: function onChange() {},
    onKeyDown: function onKeyDown() {}
  }, args));
};

exports.Default = Default;

var Disabled = function Disabled(args) {
  return /*#__PURE__*/_react.default.createElement(_.default, (0, _extends2.default)({
    disabled: true,
    size: "lg",
    defaultValue: "A default value",
    labelText: "Search",
    closeButtonLabelText: "Clear search input",
    id: "search-1",
    onChange: function onChange() {},
    onKeyDown: function onKeyDown() {}
  }, args));
};

exports.Disabled = Disabled;

var Expandable = function Expandable(args) {
  return /*#__PURE__*/_react.default.createElement(_carbonComponentsReact.ExpandableSearch, (0, _extends2.default)({
    size: "lg",
    labelText: "Search",
    closeButtonLabelText: "Clear search input",
    id: "search-expandable-1",
    onChange: function onChange() {},
    onKeyDown: function onKeyDown() {}
  }, args));
};

exports.Expandable = Expandable;

var WithLayer = function WithLayer(args) {
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_.default, (0, _extends2.default)({
    size: "lg",
    defaultValue: "First Layer",
    labelText: "Search",
    closeButtonLabelText: "Clear search input",
    id: "search-1",
    onChange: function onChange() {},
    onKeyDown: function onKeyDown() {}
  }, args)), /*#__PURE__*/_react.default.createElement(_Layer.Layer, null, /*#__PURE__*/_react.default.createElement(_.default, (0, _extends2.default)({
    size: "lg",
    defaultValue: "Second Layer",
    labelText: "Search",
    closeButtonLabelText: "Clear search input",
    id: "search-1",
    onChange: function onChange() {},
    onKeyDown: function onKeyDown() {}
  }, args)), /*#__PURE__*/_react.default.createElement(_Layer.Layer, null, /*#__PURE__*/_react.default.createElement(_.default, (0, _extends2.default)({
    size: "lg",
    defaultValue: "Third Layer",
    labelText: "Search",
    closeButtonLabelText: "Clear search input",
    id: "search-1",
    onChange: function onChange() {},
    onKeyDown: function onKeyDown() {}
  }, args)))));
};

exports.WithLayer = WithLayer;

var ExpandableWithLayer = function ExpandableWithLayer(args) {
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_carbonComponentsReact.ExpandableSearch, (0, _extends2.default)({
    size: "lg",
    labelText: "First Layer",
    closeButtonLabelText: "Clear search input",
    id: "search-expandable-1",
    onChange: function onChange() {},
    onKeyDown: function onKeyDown() {}
  }, args)), /*#__PURE__*/_react.default.createElement(_Layer.Layer, null, /*#__PURE__*/_react.default.createElement(_carbonComponentsReact.ExpandableSearch, (0, _extends2.default)({
    size: "lg",
    labelText: "Second Layer",
    closeButtonLabelText: "Clear search input",
    id: "search-expandable-1",
    onChange: function onChange() {},
    onKeyDown: function onKeyDown() {}
  }, args)), /*#__PURE__*/_react.default.createElement(_Layer.Layer, null, /*#__PURE__*/_react.default.createElement(_carbonComponentsReact.ExpandableSearch, (0, _extends2.default)({
    size: "lg",
    labelText: "Third Layer",
    closeButtonLabelText: "Clear search input",
    id: "search-expandable-1",
    onChange: function onChange() {},
    onKeyDown: function onKeyDown() {}
  }, args)))));
};

exports.ExpandableWithLayer = ExpandableWithLayer;
