// carbon-components-react/lib/components/Notification/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ToastNotification = exports.NotificationTextDetails = exports.NotificationButton = exports.NotificationActionButton = exports.InlineNotification = exports.ActionableNotification = void 0;

var _Notification = require("./next/Notification");

var _Notification2 = require("./Notification");

var _ComponentToggle = require("../../internal/ComponentToggle");

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
var NotificationActionButton = (0, _ComponentToggle.createComponentToggle)({
  name: 'NotificationActionButton',
  next: _Notification.NotificationActionButton,
  classic: _Notification2.NotificationActionButton
});
exports.NotificationActionButton = NotificationActionButton;
var NotificationTextDetails = (0, _ComponentToggle.createComponentToggle)({
  name: 'NotificationTextDetails',
  classic: _Notification2.NotificationTextDetails
});
exports.NotificationTextDetails = NotificationTextDetails;
var NotificationButton = (0, _ComponentToggle.createComponentToggle)({
  name: 'NotificationButton',
  next: _Notification.NotificationButton,
  classic: _Notification2.NotificationButton
});
exports.NotificationButton = NotificationButton;
var ToastNotification = (0, _ComponentToggle.createComponentToggle)({
  name: 'ToastNotification',
  next: _Notification.ToastNotification,
  classic: _Notification2.ToastNotification
});
exports.ToastNotification = ToastNotification;
var InlineNotification = (0, _ComponentToggle.createComponentToggle)({
  name: 'InlineNotification',
  next: _Notification.InlineNotification,
  classic: _Notification2.InlineNotification
});
exports.InlineNotification = InlineNotification;
var ActionableNotification = (0, _ComponentToggle.createComponentToggle)({
  name: 'ActionableNotification',
  next: _Notification.ActionableNotification,
  classic: null
});
exports.ActionableNotification = ActionableNotification;
