// carbon-components-react/es/components/AspectRatio/next/AspectRatio.stories.js
/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import '../AspectRatio-story.scss';
import React from 'react';
import { Grid, Column } from '../../Grid';
import { AspectRatio } from '../';
import mdx from '../AspectRatio.mdx';
export default {
  title: 'Components/AspectRatio',
  component: AspectRatio,
  decorators: [function (Story) {
    return /*#__PURE__*/React.createElement("div", {
      className: "aspect-ratio-story"
    }, /*#__PURE__*/React.createElement(Story, null));
  }],
  parameters: {
    docs: {
      page: mdx
    }
  }
};
export var AspectRatioStory = function AspectRatioStory() {
  return /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Column, {
    sm: 1,
    md: 2,
    lg: 4
  }, /*#__PURE__*/React.createElement(AspectRatio, {
    ratio: "1x1"
  }, "Content")), /*#__PURE__*/React.createElement(Column, {
    sm: 1,
    md: 2,
    lg: 4
  }, /*#__PURE__*/React.createElement(AspectRatio, {
    ratio: "1x1"
  }, "Content")), /*#__PURE__*/React.createElement(Column, {
    sm: 1,
    md: 2,
    lg: 4
  }, /*#__PURE__*/React.createElement(AspectRatio, {
    ratio: "1x1"
  }, "Content")), /*#__PURE__*/React.createElement(Column, {
    sm: 1,
    md: 2,
    lg: 4
  }, /*#__PURE__*/React.createElement(AspectRatio, {
    ratio: "1x1"
  }, "Content")));
};
AspectRatioStory.storyName = 'AspectRatio';
export var Playground = {
  argTypes: {
    as: {
      control: {
        type: null
      }
    },
    children: {
      control: {
        type: null
      }
    },
    className: {
      control: {
        type: null
      }
    },
    ratio: {
      control: {
        type: 'select'
      },
      defaultValue: '1x1',
      options: ['16x9', '9x16', '2x1', '1x2', '4x3', '3x4', '1x1']
    }
  },
  render: function render(_ref) {
    var ratio = _ref.ratio;
    return /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Column, {
      sm: 1,
      md: 2,
      lg: 4
    }, /*#__PURE__*/React.createElement(AspectRatio, {
      ratio: ratio
    }, "Content")), /*#__PURE__*/React.createElement(Column, {
      sm: 1,
      md: 2,
      lg: 4
    }, /*#__PURE__*/React.createElement(AspectRatio, {
      ratio: ratio
    }, "Content")), /*#__PURE__*/React.createElement(Column, {
      sm: 1,
      md: 2,
      lg: 4
    }, /*#__PURE__*/React.createElement(AspectRatio, {
      ratio: ratio
    }, "Content")), /*#__PURE__*/React.createElement(Column, {
      sm: 1,
      md: 2,
      lg: 4
    }, /*#__PURE__*/React.createElement(AspectRatio, {
      ratio: ratio
    }, "Content")));
  }
};
