// carbon-components-react/es/components/Grid/next/FlexGrid.stories.js
import './FlexGrid.stories.scss';
import React from 'react';
import { FlexGrid, Row, Column } from '../';
import mdx from './FlexGrid.mdx';
export default {
  title: 'Components/FlexGrid',
  component: FlexGrid,
  subcomponents: {
    Row: Row,
    Column: Column
  },
  decorators: [function (storyFn) {
    return /*#__PURE__*/React.createElement("div", {
      id: "templates"
    }, storyFn());
  }],
  parameters: {
    docs: {
      page: mdx
    }
  }
};

function DemoContent(_ref) {
  var children = _ref.children;
  return /*#__PURE__*/React.createElement("div", {
    className: "outside"
  }, /*#__PURE__*/React.createElement("div", {
    className: "inside"
  }, children));
}

export var AutoColumns = function AutoColumns() {
  return /*#__PURE__*/React.createElement(FlexGrid, null, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "Span 25%")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "Span 25%")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "Span 25%")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "Span 25%"))));
};
export var ResponsiveGrid = function ResponsiveGrid() {
  return /*#__PURE__*/React.createElement(FlexGrid, null, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Column, {
    sm: 2,
    md: 4,
    lg: 6
  }, /*#__PURE__*/React.createElement(DemoContent, null, /*#__PURE__*/React.createElement("p", null, "Small: Span 2 of 4"), /*#__PURE__*/React.createElement("p", null, "Medium: Span 4 of 8"), /*#__PURE__*/React.createElement("p", null, "Large: Span 6 of 16"))), /*#__PURE__*/React.createElement(Column, {
    sm: 2,
    md: 2,
    lg: 3
  }, /*#__PURE__*/React.createElement(DemoContent, null, /*#__PURE__*/React.createElement("p", null, "Small: Span 2 of 4"), /*#__PURE__*/React.createElement("p", null, "Medium: Span 2 of 8"), /*#__PURE__*/React.createElement("p", null, "Large: Span 3 of 16"))), /*#__PURE__*/React.createElement(Column, {
    sm: 0,
    md: 2,
    lg: 3
  }, /*#__PURE__*/React.createElement(DemoContent, null, /*#__PURE__*/React.createElement("p", null, "Small: Span 0 of 4"), /*#__PURE__*/React.createElement("p", null, "Medium: Span 2 of 8"), /*#__PURE__*/React.createElement("p", null, "Large: Span 3 of 16")))));
};
export var Offset = function Offset() {
  return /*#__PURE__*/React.createElement(FlexGrid, null, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Column, {
    sm: {
      span: 1,
      offset: 3
    }
  }, /*#__PURE__*/React.createElement(DemoContent, null, "Small: offset 3")), /*#__PURE__*/React.createElement(Column, {
    sm: {
      span: 2,
      offset: 2
    }
  }, /*#__PURE__*/React.createElement(DemoContent, null, "Small: offset 2")), /*#__PURE__*/React.createElement(Column, {
    sm: {
      span: 3,
      offset: 1
    }
  }, /*#__PURE__*/React.createElement(DemoContent, null, "Small: offset 1")), /*#__PURE__*/React.createElement(Column, {
    sm: {
      span: 4,
      offset: 0
    }
  }, /*#__PURE__*/React.createElement(DemoContent, null, "Small: offset 0"))));
};
export var Condensed = function Condensed() {
  return /*#__PURE__*/React.createElement(FlexGrid, {
    condensed: true
  }, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4"))));
};
export var CondensedColumns = function CondensedColumns() {
  return /*#__PURE__*/React.createElement(FlexGrid, null, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4"))), /*#__PURE__*/React.createElement(Row, {
    condensed: true
  }, /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4"))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4"))));
};
export var Narrow = function Narrow() {
  return /*#__PURE__*/React.createElement(FlexGrid, {
    narrow: true
  }, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4"))));
};
export var NarrowColumns = function NarrowColumns() {
  return /*#__PURE__*/React.createElement(FlexGrid, null, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4"))), /*#__PURE__*/React.createElement(Row, {
    narrow: true
  }, /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4"))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4"))));
};
export var FullWidth = function FullWidth() {
  return /*#__PURE__*/React.createElement(FlexGrid, {
    fullWidth: true
  }, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4"))));
};
export var MixedGridModes = function MixedGridModes() {
  return /*#__PURE__*/React.createElement(FlexGrid, null, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "Wide")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4"))), /*#__PURE__*/React.createElement(Row, {
    narrow: true
  }, /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "Narrow")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4"))), /*#__PURE__*/React.createElement(Row, {
    condensed: true
  }, /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "Condensed")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(DemoContent, null, "1/4"))));
};
