// carbon-components-react/es/components/Grid/next/Grid.stories.js
/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import './Grid.stories.scss';
import React from 'react';
import { Grid, Column, ColumnHang } from '../../Grid';
import mdx from './Grid.mdx';
export default {
  title: 'Elements/Grid',
  component: Grid,
  subcomponents: {
    Column: Column
  },
  parameters: {
    controls: {
      hideNoControlsWarning: true
    },
    docs: {
      page: mdx
    }
  },
  decorators: [function (Story) {
    return /*#__PURE__*/React.createElement("div", {
      className: "sb-css-grid-container"
    }, /*#__PURE__*/React.createElement(Story, null));
  }]
};
export var Default = function Default() {
  return /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Column, {
    sm: 4
  }), /*#__PURE__*/React.createElement(Column, {
    sm: 4
  }), /*#__PURE__*/React.createElement(Column, {
    sm: 4
  }), /*#__PURE__*/React.createElement(Column, {
    sm: 4
  }));
};
export var Narrow = function Narrow() {
  return /*#__PURE__*/React.createElement(Grid, {
    narrow: true
  }, /*#__PURE__*/React.createElement(Column, {
    sm: 4
  }), /*#__PURE__*/React.createElement(Column, {
    sm: 4
  }), /*#__PURE__*/React.createElement(Column, {
    sm: 4
  }), /*#__PURE__*/React.createElement(Column, {
    sm: 4
  }));
};
export var Condensed = function Condensed() {
  return /*#__PURE__*/React.createElement(Grid, {
    condensed: true
  }, /*#__PURE__*/React.createElement(Column, {
    sm: 4
  }), /*#__PURE__*/React.createElement(Column, {
    sm: 4
  }), /*#__PURE__*/React.createElement(Column, {
    sm: 4
  }), /*#__PURE__*/React.createElement(Column, {
    sm: 4
  }));
};
export var FullWidth = function FullWidth() {
  return /*#__PURE__*/React.createElement(Grid, {
    fullWidth: true
  }, /*#__PURE__*/React.createElement(Column, {
    sm: 4
  }), /*#__PURE__*/React.createElement(Column, {
    sm: 4
  }), /*#__PURE__*/React.createElement(Column, {
    sm: 4
  }), /*#__PURE__*/React.createElement(Column, {
    sm: 4
  }));
};
export var Responsive = function Responsive() {
  return /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Column, {
    sm: 2,
    md: 4,
    lg: 6
  }, /*#__PURE__*/React.createElement("p", null, "Small: Span 2 of 4"), /*#__PURE__*/React.createElement("p", null, "Medium: Span 4 of 8"), /*#__PURE__*/React.createElement("p", null, "Large: Span 6 of 16")), /*#__PURE__*/React.createElement(Column, {
    sm: 2,
    md: 2,
    lg: 3
  }, /*#__PURE__*/React.createElement("p", null, "Small: Span 2 of 4"), /*#__PURE__*/React.createElement("p", null, "Medium: Span 2 of 8"), /*#__PURE__*/React.createElement("p", null, "Large: Span 3 of 16")), /*#__PURE__*/React.createElement(Column, {
    sm: 0,
    md: 2,
    lg: 3
  }, /*#__PURE__*/React.createElement("p", null, "Small: Span 0 of 4"), /*#__PURE__*/React.createElement("p", null, "Medium: Span 2 of 8"), /*#__PURE__*/React.createElement("p", null, "Large: Span 3 of 16")), /*#__PURE__*/React.createElement(Column, {
    sm: 0,
    md: 0,
    lg: 4
  }, /*#__PURE__*/React.createElement("p", null, "Small: Span 0 of 4"), /*#__PURE__*/React.createElement("p", null, "Medium: Span 0 of 8"), /*#__PURE__*/React.createElement("p", null, "Large: Span 4 of 16")), /*#__PURE__*/React.createElement(Column, {
    sm: "25%",
    md: "50%",
    lg: "75%"
  }, /*#__PURE__*/React.createElement("p", null, "Small: Span 25%"), /*#__PURE__*/React.createElement("p", null, "Medium: Span 50%"), /*#__PURE__*/React.createElement("p", null, "Large: Span 75%")));
};
export var Subgrid = function Subgrid() {
  return /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Column, {
    sm: 2,
    md: 4,
    lg: 3
  }, /*#__PURE__*/React.createElement("p", null, "Small: Span 2 of 4"), /*#__PURE__*/React.createElement("p", null, "Medium: Span 4 of 8"), /*#__PURE__*/React.createElement("p", null, "Large: Span 3 of 16")), /*#__PURE__*/React.createElement(Column, {
    sm: 2,
    md: 4,
    lg: 10
  }, /*#__PURE__*/React.createElement("p", null, "Small: Span 2 of 4"), /*#__PURE__*/React.createElement("p", null, "Medium: Span 4 of 8"), /*#__PURE__*/React.createElement("p", null, "Large: Span 10 of 16"), /*#__PURE__*/React.createElement(Grid, {
    className: "example"
  }, /*#__PURE__*/React.createElement(Column, {
    sm: 1,
    md: 1,
    lg: 2
  }, /*#__PURE__*/React.createElement("p", null, "sm=", 1), " ", /*#__PURE__*/React.createElement("p", null, "md=", 1), " ", /*#__PURE__*/React.createElement("p", null, "lg=", 2)), /*#__PURE__*/React.createElement(Column, {
    sm: 1,
    md: 1,
    lg: 2
  }, /*#__PURE__*/React.createElement("p", null, "sm=", 1), " ", /*#__PURE__*/React.createElement("p", null, "md=", 1), " ", /*#__PURE__*/React.createElement("p", null, "lg=", 2)), /*#__PURE__*/React.createElement(Column, {
    sm: 0,
    md: 1,
    lg: 1
  }, /*#__PURE__*/React.createElement("p", null, "sm=", 0), " ", /*#__PURE__*/React.createElement("p", null, "md=", 1), " ", /*#__PURE__*/React.createElement("p", null, "lg=", 1)), /*#__PURE__*/React.createElement(Column, {
    sm: 0,
    md: 1,
    lg: 1
  }, /*#__PURE__*/React.createElement("p", null, "sm=", 0), " ", /*#__PURE__*/React.createElement("p", null, "md=", 1), " ", /*#__PURE__*/React.createElement("p", null, "lg=", 1)), /*#__PURE__*/React.createElement(Column, {
    sm: 0,
    md: 0,
    lg: 1
  }, /*#__PURE__*/React.createElement("p", null, "sm=", 0), " ", /*#__PURE__*/React.createElement("p", null, "md=", 0), " ", /*#__PURE__*/React.createElement("p", null, "lg=", 1)), /*#__PURE__*/React.createElement(Column, {
    sm: 0,
    md: 0,
    lg: 1
  }, /*#__PURE__*/React.createElement("p", null, "sm=", 0), " ", /*#__PURE__*/React.createElement("p", null, "md=", 0), " ", /*#__PURE__*/React.createElement("p", null, "lg=", 1)), /*#__PURE__*/React.createElement(Column, {
    sm: 0,
    md: 0,
    lg: 1
  }, /*#__PURE__*/React.createElement("p", null, "sm=", 0), " ", /*#__PURE__*/React.createElement("p", null, "md=", 0), " ", /*#__PURE__*/React.createElement("p", null, "lg=", 1)), /*#__PURE__*/React.createElement(Column, {
    sm: 0,
    md: 0,
    lg: 1
  }, /*#__PURE__*/React.createElement("p", null, "sm=", 0), " ", /*#__PURE__*/React.createElement("p", null, "md=", 0), " ", /*#__PURE__*/React.createElement("p", null, "lg=", 1)))), /*#__PURE__*/React.createElement(Column, {
    sm: 0,
    md: 0,
    lg: 3
  }, /*#__PURE__*/React.createElement("p", null, "Small: Span 0 of 4"), /*#__PURE__*/React.createElement("p", null, "Medium: Span 0 of 8"), /*#__PURE__*/React.createElement("p", null, "Large: Span 3 of 16")));
};
export var MixedGridModes = function MixedGridModes() {
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Column, {
    span: 8
  }, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Column, {
    span: 8
  }, /*#__PURE__*/React.createElement(Grid, {
    narrow: true
  }, /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(ColumnHang, null, "Text")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(ColumnHang, null, "Text")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(ColumnHang, null, "Text")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(ColumnHang, null, "Text")), /*#__PURE__*/React.createElement(Column, {
    span: 4
  }, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Column, null, "Text"), /*#__PURE__*/React.createElement(Column, null, "Text"), /*#__PURE__*/React.createElement(Column, {
    span: 2
  }, /*#__PURE__*/React.createElement(Grid, {
    condensed: true
  }, /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(ColumnHang, null, "Text")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(ColumnHang, null, "Text"))))))))))), /*#__PURE__*/React.createElement(Grid, {
    narrow: true
  }, /*#__PURE__*/React.createElement(Column, {
    span: 8
  }, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Column, {
    span: 4
  }), /*#__PURE__*/React.createElement(Column, {
    span: 4
  }, /*#__PURE__*/React.createElement(Grid, {
    narrow: true
  }, /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(ColumnHang, null, "Text")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(ColumnHang, null, "Text")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(ColumnHang, null, "Text")), /*#__PURE__*/React.createElement(Column, null, /*#__PURE__*/React.createElement(ColumnHang, null, "Text"))))))));
};
export var GridStartEnd = function GridStartEnd() {
  return /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Column, {
    sm: {
      span: 1,
      start: 4
    },
    md: {
      span: 2,
      start: 7
    },
    lg: {
      span: 4,
      start: 13
    }
  }, "span, start"), /*#__PURE__*/React.createElement(Column, {
    sm: {
      span: 2,
      end: 5
    },
    md: {
      span: 4,
      end: 9
    },
    lg: {
      span: 8,
      end: 17
    }
  }, "span, end"), /*#__PURE__*/React.createElement(Column, {
    sm: {
      start: 1,
      end: 4
    },
    md: {
      start: 3,
      end: 9
    },
    lg: {
      start: 5,
      end: 17
    }
  }, "start, end"));
};
export var Offset = function Offset() {
  return /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Column, {
    sm: {
      span: 1,
      offset: 3
    },
    md: {
      span: 2,
      offset: 6
    },
    lg: {
      span: 4,
      offset: 12
    }
  }), /*#__PURE__*/React.createElement(Column, {
    sm: {
      span: 2,
      offset: 2
    },
    md: {
      span: 4,
      offset: 4
    },
    lg: {
      span: 8,
      offset: 8
    }
  }), /*#__PURE__*/React.createElement(Column, {
    sm: {
      span: 3,
      offset: 1
    },
    md: {
      span: 6,
      offset: 2
    },
    lg: {
      span: 12,
      offset: 4
    }
  }), /*#__PURE__*/React.createElement(Column, {
    sm: {
      span: 4
    },
    md: {
      span: 8
    },
    lg: {
      span: 16
    }
  }), /*#__PURE__*/React.createElement(Column, {
    sm: {
      span: '25%',
      offset: 1
    },
    md: {
      span: '50%',
      offset: 2
    },
    lg: {
      span: '75%',
      offset: 4
    }
  }));
};
