// carbon-components-react/lib/components/ProgressBar/ProgressBar-story.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports._ProgressBar = exports.Playground = exports.Indeterminate = exports.Example = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _addonKnobs = require("@storybook/addon-knobs");

var _ProgressBar2 = _interopRequireDefault(require("../ProgressBar"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

/**
 * Copyright IBM Corp. 2021
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
var sizes = {
  'Small  (small)': 'small',
  'Big (big) - default': 'big'
};
var statuses = {
  'Active  (active) - default': 'active',
  'Finished (finished)': 'finished',
  'Error (error)': 'error'
};
var types = {
  'Default  (default)': 'default',
  'Inline (inline)': 'inline',
  'indented (indented)': 'indented'
};

var props = function props() {
  return {
    helperText: (0, _addonKnobs.text)('Helper text (helperText)', 'Optional helper text'),
    hideLabel: (0, _addonKnobs.boolean)('Hide the label (hideLabel)', false),
    label: (0, _addonKnobs.text)('Label text (label)', 'Progress bar label'),
    max: (0, _addonKnobs.number)('Maximum value (max)', 100),
    size: (0, _addonKnobs.select)('Size (size)', sizes, 'big'),
    status: (0, _addonKnobs.select)('Status (status)', statuses, 'active'),
    type: (0, _addonKnobs.select)('Type (type)', types, 'default'),
    value: (0, _addonKnobs.number)('Current value (value)', 75)
  };
};

var _default = {
  title: 'Experimental/unstable_ProgressBar',
  component: _ProgressBar2.default,
  decorators: [_addonKnobs.withKnobs]
};
exports.default = _default;

var _ProgressBar = function _ProgressBar() {
  return /*#__PURE__*/_react.default.createElement(_ProgressBar2.default, {
    label: "Progress bar label",
    helperText: "Optional helper text",
    value: 75
  });
};

exports._ProgressBar = _ProgressBar;
_ProgressBar.storyName = 'ProgressBar';

var Indeterminate = function Indeterminate() {
  return /*#__PURE__*/_react.default.createElement(_ProgressBar2.default, {
    label: "Progress bar label",
    helperText: "Optional helper text"
  });
};

exports.Indeterminate = Indeterminate;

var Example = function Example() {
  var size = 728;

  var _useState = (0, _react.useState)(0),
      _useState2 = (0, _slicedToArray2.default)(_useState, 2),
      progress = _useState2[0],
      setProgress = _useState2[1];

  (0, _react.useEffect)(function () {
    setTimeout(function () {
      var interval = setInterval(function () {
        setProgress(function (currentProgress) {
          var advancement = Math.random() * 8;

          if (currentProgress + advancement < size) {
            return currentProgress + advancement;
          } else {
            clearInterval(interval);
            return size;
          }
        });
      }, 50);
    }, 3000);
  }, []);
  var running = progress > 0;
  var helperText = running ? "".concat(progress.toFixed(1), "MB of ").concat(size, "MB") : 'Fetching assets...';

  if (progress >= size) {
    helperText = 'Done';
  }

  return /*#__PURE__*/_react.default.createElement(_ProgressBar2.default, {
    value: running ? progress : null,
    max: size,
    status: progress === size ? 'finished' : 'active',
    label: "Export data",
    helperText: helperText
  });
};

exports.Example = Example;

var Playground = function Playground() {
  return /*#__PURE__*/_react.default.createElement(_ProgressBar2.default, props());
};

exports.Playground = Playground;
