// carbon-components-react/es/components/Toggletip/next/Toggletip.stories.js
/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import { Information16 } from '@carbon/icons-react';
import React from 'react';
import { default as Button } from '../../Button';
import { default as Link } from '../../Link';
import { ToggletipLabel, Toggletip, ToggletipButton, ToggletipContent, ToggletipActions } from '../../Toggletip';
import mdx from './Toggletip.mdx';
export default {
  title: 'Components/Toggletip',
  component: Toggletip,
  subcomponents: {
    ToggletipLabel: ToggletipLabel,
    ToggletipButton: ToggletipButton,
    ToggletipContent: ToggletipContent,
    ToggletipActions: ToggletipActions
  },
  argTypes: {
    as: {
      table: {
        disable: true
      }
    },
    children: {
      table: {
        disable: true
      }
    },
    className: {
      table: {
        disable: true
      }
    }
  },
  parameters: {
    docs: {
      page: mdx
    }
  }
};
export var Default = function Default() {
  return /*#__PURE__*/React.createElement("div", {
    style: {
      display: 'flex',
      justifyContent: 'center'
    }
  }, /*#__PURE__*/React.createElement(ToggletipLabel, null, "Toggletip label"), /*#__PURE__*/React.createElement(Toggletip, {
    defaultOpen: true
  }, /*#__PURE__*/React.createElement(ToggletipButton, {
    label: "Show information"
  }, /*#__PURE__*/React.createElement(Information16, null)), /*#__PURE__*/React.createElement(ToggletipContent, null, /*#__PURE__*/React.createElement("p", null, "Lorem ipsum dolor sit amet, di os consectetur adipiscing elit, sed do eiusmod tempor incididunt ut fsil labore et dolore magna aliqua."), /*#__PURE__*/React.createElement(ToggletipActions, null, /*#__PURE__*/React.createElement(Link, {
    href: "#"
  }, "Link action"), /*#__PURE__*/React.createElement(Button, {
    size: "sm"
  }, "Button")))));
};

var PlaygroundStory = function PlaygroundStory(controls) {
  var align = controls.align;
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(ToggletipLabel, null, "Toggletip label"), /*#__PURE__*/React.createElement(Toggletip, {
    align: align,
    defaultOpen: true
  }, /*#__PURE__*/React.createElement(ToggletipButton, {
    label: "Show information"
  }, /*#__PURE__*/React.createElement(Information16, null)), /*#__PURE__*/React.createElement(ToggletipContent, null, /*#__PURE__*/React.createElement("p", null, "Lorem ipsum dolor sit amet, di os consectetur adipiscing elit, sed do eiusmod tempor incididunt ut fsil labore et dolore magna aliqua."), /*#__PURE__*/React.createElement(ToggletipActions, null, /*#__PURE__*/React.createElement(Link, {
    href: "#"
  }, "Link action"), /*#__PURE__*/React.createElement(Button, {
    size: "sm"
  }, "Button")))));
};

export var Playground = PlaygroundStory.bind({});
Playground.argTypes = {
  align: {
    options: ['top', 'bottom', 'left', 'right'],
    control: {
      type: 'select'
    }
  }
};
Playground.story = {
  decorators: [function (story) {
    return /*#__PURE__*/React.createElement("div", {
      style: {
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center'
      }
    }, story());
  }]
};
