// carbon-components-react/lib/components/MultiSelect/next/MultiSelect.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports._FilterableWithLayer = exports._Filterable = exports.WithLayer = exports.WithInitialSelectedItems = exports.Default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _ = _interopRequireDefault(require("../"));

var _FilterableMultiSelect = _interopRequireDefault(require("./FilterableMultiSelect"));

var _Layer = require("../../Layer");

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
var _default = {
  title: 'Components/MultiSelect',
  component: _.default,
  subcomponents: {
    'MultiSelect.Filterable': _.default.Filterable
  },
  argTypes: {
    size: {
      options: ['sm', 'md', 'lg'],
      control: {
        type: 'select'
      }
    }
  }
};
exports.default = _default;
var items = [{
  id: 'downshift-1-item-0',
  text: 'Option 1'
}, {
  id: 'downshift-1-item-1',
  text: 'Option 2'
}, {
  id: 'downshift-1-item-2',
  text: 'Option 3 - a disabled item',
  disabled: true
}, {
  id: 'downshift-1-item-3',
  text: 'Option 4'
}, {
  id: 'downshift-1-item-4',
  text: 'An example option that is really long to show what should be done to handle long text'
}, {
  id: 'downshift-1-item-5',
  text: 'Option 5'
}];

var Default = function Default(args) {
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      width: 300
    }
  }, /*#__PURE__*/_react.default.createElement(_.default, (0, _extends2.default)({
    label: "Multiselect Label",
    id: "carbon-multiselect-example",
    titleText: "Multiselect title",
    helperText: "This is helper text",
    items: items,
    itemToString: function itemToString(item) {
      return item ? item.text : '';
    },
    selectionFeedback: "top-after-reopen"
  }, args)));
};

exports.Default = Default;

var WithInitialSelectedItems = function WithInitialSelectedItems(args) {
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      width: 300
    }
  }, /*#__PURE__*/_react.default.createElement(_.default, (0, _extends2.default)({
    id: "carbon-multiselect-example-2",
    titleText: "Multiselect title",
    helperText: "This is helper text",
    items: items,
    itemToString: function itemToString(item) {
      return item ? item.text : '';
    },
    initialSelectedItems: [items[0], items[1]],
    selectionFeedback: "top-after-reopen"
  }, args)));
};

exports.WithInitialSelectedItems = WithInitialSelectedItems;

var _Filterable = function _Filterable(args) {
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      width: 300
    }
  }, /*#__PURE__*/_react.default.createElement(_FilterableMultiSelect.default, (0, _extends2.default)({
    id: "carbon-multiselect-example-3",
    titleText: "Multiselect title",
    helperText: "This is helper text",
    items: items,
    itemToString: function itemToString(item) {
      return item ? item.text : '';
    },
    selectionFeedback: "top-after-reopen"
  }, args)));
};

exports._Filterable = _Filterable;

var WithLayer = function WithLayer(args) {
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      width: 300
    }
  }, /*#__PURE__*/_react.default.createElement(_.default, (0, _extends2.default)({
    label: "First Layer",
    id: "carbon-multiselect-example",
    titleText: "Multiselect title",
    helperText: "This is helper text",
    items: items,
    itemToString: function itemToString(item) {
      return item ? item.text : '';
    },
    selectionFeedback: "top-after-reopen"
  }, args)), /*#__PURE__*/_react.default.createElement(_Layer.Layer, null, /*#__PURE__*/_react.default.createElement(_.default, (0, _extends2.default)({
    label: "Second Layer",
    id: "carbon-multiselect-example",
    titleText: "Multiselect title",
    helperText: "This is helper text",
    items: items,
    itemToString: function itemToString(item) {
      return item ? item.text : '';
    },
    selectionFeedback: "top-after-reopen"
  }, args)), /*#__PURE__*/_react.default.createElement(_Layer.Layer, null, /*#__PURE__*/_react.default.createElement(_.default, (0, _extends2.default)({
    label: "Third Layer",
    id: "carbon-multiselect-example",
    titleText: "Multiselect title",
    helperText: "This is helper text",
    items: items,
    itemToString: function itemToString(item) {
      return item ? item.text : '';
    },
    selectionFeedback: "top-after-reopen"
  }, args)))));
};

exports.WithLayer = WithLayer;

var _FilterableWithLayer = function _FilterableWithLayer(args) {
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      width: 300
    }
  }, /*#__PURE__*/_react.default.createElement(_FilterableMultiSelect.default, (0, _extends2.default)({
    id: "carbon-multiselect-example-3",
    titleText: "First Layer",
    helperText: "This is helper text",
    items: items,
    itemToString: function itemToString(item) {
      return item ? item.text : '';
    },
    selectionFeedback: "top-after-reopen"
  }, args)), /*#__PURE__*/_react.default.createElement(_Layer.Layer, null, /*#__PURE__*/_react.default.createElement(_FilterableMultiSelect.default, (0, _extends2.default)({
    id: "carbon-multiselect-example-3",
    titleText: "Second Layer",
    helperText: "This is helper text",
    items: items,
    itemToString: function itemToString(item) {
      return item ? item.text : '';
    },
    selectionFeedback: "top-after-reopen"
  }, args)), /*#__PURE__*/_react.default.createElement(_Layer.Layer, null, /*#__PURE__*/_react.default.createElement(_FilterableMultiSelect.default, (0, _extends2.default)({
    id: "carbon-multiselect-example-3",
    titleText: "Third Layer",
    helperText: "This is helper text",
    items: items,
    itemToString: function itemToString(item) {
      return item ? item.text : '';
    },
    selectionFeedback: "top-after-reopen"
  }, args)))));
};

exports._FilterableWithLayer = _FilterableWithLayer;
