// carbon-components-react/lib/components/FormGroup/next/FormGroup.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _FormGroup = _interopRequireDefault(require("../FormGroup"));

var _TextInput = _interopRequireDefault(require("../../TextInput"));

var _RadioButtonGroup = _interopRequireDefault(require("../../RadioButtonGroup"));

var _RadioButton = _interopRequireDefault(require("../../RadioButton"));

var _Button = _interopRequireDefault(require("../../Button"));

var _Stack = require("../../Stack");

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
var _default = {
  title: 'Components/FormGroup',
  component: _FormGroup.default,
  argTypes: {
    disabled: {
      control: {
        type: 'boolean'
      },
      defaultValue: false
    },
    // TODO remove hasMargin in v11
    hasMargin: {
      table: {
        disable: true
      }
    },
    legendText: {
      control: {
        type: 'text'
      },
      defaultValue: 'FormGroup Legend'
    },
    legendId: {
      control: {
        type: 'text'
      },
      defaultValue: 'formgroup-legend-id'
    },
    children: {
      control: false
    }
  }
};
exports.default = _default;

var Default = function Default(args) {
  return /*#__PURE__*/_react.default.createElement(_FormGroup.default, (0, _extends2.default)({
    style: {
      maxWidth: '400px'
    }
  }, args), /*#__PURE__*/_react.default.createElement(_Stack.Stack, {
    gap: 7
  }, /*#__PURE__*/_react.default.createElement(_TextInput.default, {
    id: "one",
    labelText: "First Name"
  }), /*#__PURE__*/_react.default.createElement(_TextInput.default, {
    id: "two",
    labelText: "Last Name"
  }), /*#__PURE__*/_react.default.createElement(_RadioButtonGroup.default, {
    legendText: "Radio button heading",
    name: "radio-button-group",
    defaultSelected: "radio-1"
  }, /*#__PURE__*/_react.default.createElement(_RadioButton.default, {
    labelText: "Option 1",
    value: "radio-1",
    id: "radio-1"
  }), /*#__PURE__*/_react.default.createElement(_RadioButton.default, {
    labelText: "Option 2",
    value: "radio-2",
    id: "radio-2"
  }), /*#__PURE__*/_react.default.createElement(_RadioButton.default, {
    labelText: "Option 3",
    value: "radio-3",
    id: "radio-3"
  })), /*#__PURE__*/_react.default.createElement(_Button.default, null, "Submit")));
};

exports.Default = Default;
