// carbon-components-react/lib/components/TimePickerSelect/TimePickerSelect.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _defineProperty3 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _iconsReact = require("@carbon/icons-react");

var _deprecate = _interopRequireDefault(require("../../prop-types/deprecate"));

var _usePrefix = require("../../internal/usePrefix");

var _defineProperty2;

var _excluded = ["aria-label", "children", "className", "disabled", "hideLabel", "id", "iconDescription", "labelText"];

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

var TimePickerSelect = /*#__PURE__*/function (_Component) {
  (0, _inherits2.default)(TimePickerSelect, _Component);

  var _super = _createSuper(TimePickerSelect);

  function TimePickerSelect() {
    (0, _classCallCheck2.default)(this, TimePickerSelect);
    return _super.apply(this, arguments);
  }

  (0, _createClass2.default)(TimePickerSelect, [{
    key: "render",
    value: function render() {
      var _classNames;

      var prefix = this.context;
      var _this$props = this.props,
          _this$props$ariaLabe = _this$props['aria-label'],
          ariaLabel = _this$props$ariaLabe === void 0 ? 'open list of options' : _this$props$ariaLabe,
          children = _this$props.children,
          className = _this$props.className,
          disabled = _this$props.disabled,
          _this$props$hideLabel = _this$props.hideLabel,
          hideLabel = _this$props$hideLabel === void 0 ? true : _this$props$hideLabel,
          id = _this$props.id,
          iconDescription = _this$props.iconDescription,
          labelText = _this$props.labelText,
          rest = (0, _objectWithoutProperties2.default)(_this$props, _excluded);
      var selectClasses = (0, _classnames.default)((_classNames = {}, (0, _defineProperty3.default)(_classNames, "".concat(prefix, "--select"), true), (0, _defineProperty3.default)(_classNames, "".concat(prefix, "--time-picker__select"), true), (0, _defineProperty3.default)(_classNames, className, className), _classNames));
      var labelClasses = (0, _classnames.default)("".concat(prefix, "--label"), (0, _defineProperty3.default)({}, "".concat(prefix, "--visually-hidden"), hideLabel));
      var label = labelText ? /*#__PURE__*/_react.default.createElement("label", {
        htmlFor: id,
        className: labelClasses
      }, labelText) : null;
      return /*#__PURE__*/_react.default.createElement("div", {
        className: selectClasses
      }, label, /*#__PURE__*/_react.default.createElement("select", (0, _extends2.default)({
        className: "".concat(prefix, "--select-input"),
        disabled: disabled,
        id: id,
        "aria-label": ariaLabel
      }, rest), children), /*#__PURE__*/_react.default.createElement(_iconsReact.ChevronDown16, {
        className: "".concat(prefix, "--select__arrow"),
        "aria-label": ariaLabel ? ariaLabel : iconDescription
      }));
    }
  }]);
  return TimePickerSelect;
}(_react.Component);

exports.default = TimePickerSelect;
(0, _defineProperty3.default)(TimePickerSelect, "contextType", _usePrefix.PrefixContext);
(0, _defineProperty3.default)(TimePickerSelect, "propTypes", (_defineProperty2 = {}, (0, _defineProperty3.default)(_defineProperty2, 'aria-label', _propTypes.default.string), (0, _defineProperty3.default)(_defineProperty2, "children", _propTypes.default.node), (0, _defineProperty3.default)(_defineProperty2, "className", _propTypes.default.string), (0, _defineProperty3.default)(_defineProperty2, "defaultValue", _propTypes.default.any), (0, _defineProperty3.default)(_defineProperty2, "disabled", _propTypes.default.bool), (0, _defineProperty3.default)(_defineProperty2, "hideLabel", (0, _deprecate.default)(_propTypes.default.bool, 'The `hideLabel` prop for `TimePickerSelect` is no longer needed and has ' + 'been deprecated. It will be removed in the next major release.')), (0, _defineProperty3.default)(_defineProperty2, "iconDescription", (0, _deprecate.default)(_propTypes.default.string, 'The `iconDescription` prop for `TimePickerSelect` is no longer needed and has ' + 'been deprecated. It will be removed in the next major release. Use `aria-label` instead.')), (0, _defineProperty3.default)(_defineProperty2, "id", _propTypes.default.string.isRequired), (0, _defineProperty3.default)(_defineProperty2, "labelText", _propTypes.default.node.isRequired), _defineProperty2));
(0, _defineProperty3.default)(TimePickerSelect, "defaultProps", {
  disabled: false
});
